// @paypal/paypal-server-sdk/dist/cjs/defaultConfiguration.js
"use strict";
/**
 * PayPal Server SDKLib
 *
 * This file was automatically generated by APIMATIC v3.0 ( https://www.apimatic.io ).
 */
exports.__esModule = true;
exports.DEFAULT_LOGGING_OPTIONS = exports.DEFAULT_RETRY_CONFIG = exports.DEFAULT_CONFIGURATION = void 0;
var configuration_1 = require("./configuration");
var core_1 = require("./core");
/** Default values for the configuration parameters of the client. */
exports.DEFAULT_CONFIGURATION = {
    timeout: 0,
    environment: configuration_1.Environment.Sandbox
};
/** Default values for retry configuration parameters. */
exports.DEFAULT_RETRY_CONFIG = {
    maxNumberOfRetries: 0,
    retryOnTimeout: true,
    retryInterval: 1,
    maximumRetryWaitTime: 0,
    backoffFactor: 2,
    httpStatusCodesToRetry: [408, 413, 429, 500, 502, 503, 504, 521, 522, 524],
    httpMethodsToRetry: ['GET', 'PUT']
};
/** Default values for logging options. */
exports.DEFAULT_LOGGING_OPTIONS = (0, core_1.mergeLoggingOptions)({
    logger: new core_1.NullLogger()
});
