// @paypal/paypal-server-sdk/dist/esm/models/item.js
/**
 * PayPal Server SDKLib
 *
 * This file was automatically generated by APIMATIC v3.0 ( https://www.apimatic.io ).
 */
import { lazy, object, optional, string } from '../schema';
import { itemCategorySchema } from './itemCategory';
import { moneySchema } from './money';
import { orderBillingPlanSchema } from './orderBillingPlan';
import { universalProductCodeSchema, } from './universalProductCode';
export const itemSchema = object({
    name: ['name', string()],
    unitAmount: ['unit_amount', lazy(() => moneySchema)],
    tax: ['tax', optional(lazy(() => moneySchema))],
    quantity: ['quantity', string()],
    description: ['description', optional(string())],
    sku: ['sku', optional(string())],
    url: ['url', optional(string())],
    category: ['category', optional(itemCategorySchema)],
    imageUrl: ['image_url', optional(string())],
    upc: ['upc', optional(lazy(() => universalProductCodeSchema))],
    billingPlan: ['billing_plan', optional(lazy(() => orderBillingPlanSchema))],
});
