// @paypal/paypal-server-sdk/dist/esm/models/order.js
/**
 * PayPal Server SDKLib
 *
 * This file was automatically generated by APIMATIC v3.0 ( https://www.apimatic.io ).
 */
import { array, lazy, object, optional, string } from '../schema';
import { checkoutPaymentIntentSchema, } from './checkoutPaymentIntent';
import { linkDescriptionSchema } from './linkDescription';
import { orderStatusSchema } from './orderStatus';
import { payerSchema } from './payer';
import { paymentSourceResponseSchema, } from './paymentSourceResponse';
import { purchaseUnitSchema } from './purchaseUnit';
export const orderSchema = object({
    createTime: ['create_time', optional(string())],
    updateTime: ['update_time', optional(string())],
    id: ['id', optional(string())],
    paymentSource: [
        'payment_source',
        optional(lazy(() => paymentSourceResponseSchema)),
    ],
    intent: ['intent', optional(checkoutPaymentIntentSchema)],
    payer: ['payer', optional(lazy(() => payerSchema))],
    purchaseUnits: [
        'purchase_units',
        optional(array(lazy(() => purchaseUnitSchema))),
    ],
    status: ['status', optional(orderStatusSchema)],
    links: ['links', optional(array(lazy(() => linkDescriptionSchema)))],
});
