// @paypal/paypal-server-sdk/dist/esm/models/paymentAuthorization.js
/**
 * PayPal Server SDKLib
 *
 * This file was automatically generated by APIMATIC v3.0 ( https://www.apimatic.io ).
 */
import { array, lazy, object, optional, string } from '../schema';
import { authorizationStatusSchema, } from './authorizationStatus';
import { authorizationStatusDetailsSchema, } from './authorizationStatusDetails';
import { linkDescriptionSchema } from './linkDescription';
import { moneySchema } from './money';
import { networkTransactionReferenceSchema, } from './networkTransactionReference';
import { payeeBaseSchema } from './payeeBase';
import { paymentSupplementaryDataSchema, } from './paymentSupplementaryData';
import { sellerProtectionSchema } from './sellerProtection';
export const paymentAuthorizationSchema = object({
    status: ['status', optional(authorizationStatusSchema)],
    statusDetails: [
        'status_details',
        optional(lazy(() => authorizationStatusDetailsSchema)),
    ],
    id: ['id', optional(string())],
    amount: ['amount', optional(lazy(() => moneySchema))],
    invoiceId: ['invoice_id', optional(string())],
    customId: ['custom_id', optional(string())],
    networkTransactionReference: [
        'network_transaction_reference',
        optional(lazy(() => networkTransactionReferenceSchema)),
    ],
    sellerProtection: [
        'seller_protection',
        optional(lazy(() => sellerProtectionSchema)),
    ],
    expirationTime: ['expiration_time', optional(string())],
    links: ['links', optional(array(lazy(() => linkDescriptionSchema)))],
    createTime: ['create_time', optional(string())],
    updateTime: ['update_time', optional(string())],
    supplementaryData: [
        'supplementary_data',
        optional(lazy(() => paymentSupplementaryDataSchema)),
    ],
    payee: ['payee', optional(lazy(() => payeeBaseSchema))],
});
