// @paypal/paypal-server-sdk/dist/esm/models/processorResponseCode.js
/**
 * PayPal Server SDKLib
 *
 * This file was automatically generated by APIMATIC v3.0 ( https://www.apimatic.io ).
 */
import { stringEnum } from '../schema';
/**
 * Enum for ProcessorResponseCode
 */
export var ProcessorResponseCode;
(function (ProcessorResponseCode) {
    ProcessorResponseCode["Response0000"] = "0000";
    ProcessorResponseCode["Response00N7"] = "00N7";
    ProcessorResponseCode["Response0100"] = "0100";
    ProcessorResponseCode["Response0390"] = "0390";
    ProcessorResponseCode["Response0500"] = "0500";
    ProcessorResponseCode["Response0580"] = "0580";
    ProcessorResponseCode["Response0800"] = "0800";
    ProcessorResponseCode["Response0880"] = "0880";
    ProcessorResponseCode["Response0890"] = "0890";
    ProcessorResponseCode["Response0960"] = "0960";
    ProcessorResponseCode["Response0R00"] = "0R00";
    ProcessorResponseCode["Response1000"] = "1000";
    ProcessorResponseCode["Response10Br"] = "10BR";
    ProcessorResponseCode["Response1300"] = "1300";
    ProcessorResponseCode["Response1310"] = "1310";
    ProcessorResponseCode["Response1312"] = "1312";
    ProcessorResponseCode["Response1317"] = "1317";
    ProcessorResponseCode["Response1320"] = "1320";
    ProcessorResponseCode["Response1330"] = "1330";
    ProcessorResponseCode["Response1335"] = "1335";
    ProcessorResponseCode["Response1340"] = "1340";
    ProcessorResponseCode["Response1350"] = "1350";
    ProcessorResponseCode["Response1352"] = "1352";
    ProcessorResponseCode["Response1360"] = "1360";
    ProcessorResponseCode["Response1370"] = "1370";
    ProcessorResponseCode["Response1380"] = "1380";
    ProcessorResponseCode["Response1382"] = "1382";
    ProcessorResponseCode["Response1384"] = "1384";
    ProcessorResponseCode["Response1390"] = "1390";
    ProcessorResponseCode["Response1393"] = "1393";
    ProcessorResponseCode["Response5100"] = "5100";
    ProcessorResponseCode["Response5110"] = "5110";
    ProcessorResponseCode["Response5120"] = "5120";
    ProcessorResponseCode["Response5130"] = "5130";
    ProcessorResponseCode["Response5135"] = "5135";
    ProcessorResponseCode["Response5140"] = "5140";
    ProcessorResponseCode["Response5150"] = "5150";
    ProcessorResponseCode["Response5160"] = "5160";
    ProcessorResponseCode["Response5170"] = "5170";
    ProcessorResponseCode["Response5180"] = "5180";
    ProcessorResponseCode["Response5190"] = "5190";
    ProcessorResponseCode["Response5200"] = "5200";
    ProcessorResponseCode["Response5210"] = "5210";
    ProcessorResponseCode["Response5400"] = "5400";
    ProcessorResponseCode["Response5500"] = "5500";
    ProcessorResponseCode["Response5650"] = "5650";
    ProcessorResponseCode["Response5700"] = "5700";
    ProcessorResponseCode["Response5710"] = "5710";
    ProcessorResponseCode["Response5800"] = "5800";
    ProcessorResponseCode["Response5900"] = "5900";
    ProcessorResponseCode["Response5910"] = "5910";
    ProcessorResponseCode["Response5920"] = "5920";
    ProcessorResponseCode["Response5930"] = "5930";
    ProcessorResponseCode["Response5950"] = "5950";
    ProcessorResponseCode["Response6300"] = "6300";
    ProcessorResponseCode["Response7600"] = "7600";
    ProcessorResponseCode["Response7700"] = "7700";
    ProcessorResponseCode["Response7710"] = "7710";
    ProcessorResponseCode["Response7800"] = "7800";
    ProcessorResponseCode["Response7900"] = "7900";
    ProcessorResponseCode["Response8000"] = "8000";
    ProcessorResponseCode["Response8010"] = "8010";
    ProcessorResponseCode["Response8020"] = "8020";
    ProcessorResponseCode["Response8030"] = "8030";
    ProcessorResponseCode["Response8100"] = "8100";
    ProcessorResponseCode["Response8110"] = "8110";
    ProcessorResponseCode["Response8220"] = "8220";
    ProcessorResponseCode["Response9100"] = "9100";
    ProcessorResponseCode["Response9500"] = "9500";
    ProcessorResponseCode["Response9510"] = "9510";
    ProcessorResponseCode["Response9520"] = "9520";
    ProcessorResponseCode["Response9530"] = "9530";
    ProcessorResponseCode["Response9540"] = "9540";
    ProcessorResponseCode["Response9600"] = "9600";
    ProcessorResponseCode["ResponsePcnr"] = "PCNR";
    ProcessorResponseCode["ResponsePcvv"] = "PCVV";
    ProcessorResponseCode["ResponsePp06"] = "PP06";
    ProcessorResponseCode["ResponsePprn"] = "PPRN";
    ProcessorResponseCode["ResponsePpad"] = "PPAD";
    ProcessorResponseCode["ResponsePpab"] = "PPAB";
    ProcessorResponseCode["ResponsePpae"] = "PPAE";
    ProcessorResponseCode["ResponsePpag"] = "PPAG";
    ProcessorResponseCode["ResponsePpai"] = "PPAI";
    ProcessorResponseCode["ResponsePpar"] = "PPAR";
    ProcessorResponseCode["ResponsePpau"] = "PPAU";
    ProcessorResponseCode["ResponsePpav"] = "PPAV";
    ProcessorResponseCode["ResponsePpax"] = "PPAX";
    ProcessorResponseCode["ResponsePpbg"] = "PPBG";
    ProcessorResponseCode["ResponsePpc2"] = "PPC2";
    ProcessorResponseCode["ResponsePpce"] = "PPCE";
    ProcessorResponseCode["ResponsePpco"] = "PPCO";
    ProcessorResponseCode["ResponsePpcr"] = "PPCR";
    ProcessorResponseCode["ResponsePpct"] = "PPCT";
    ProcessorResponseCode["ResponsePpcu"] = "PPCU";
    ProcessorResponseCode["ResponsePpd3"] = "PPD3";
    ProcessorResponseCode["ResponsePpdc"] = "PPDC";
    ProcessorResponseCode["ResponsePpdi"] = "PPDI";
    ProcessorResponseCode["ResponsePpdv"] = "PPDV";
    ProcessorResponseCode["ResponsePpdt"] = "PPDT";
    ProcessorResponseCode["ResponsePpef"] = "PPEF";
    ProcessorResponseCode["ResponsePpel"] = "PPEL";
    ProcessorResponseCode["ResponsePper"] = "PPER";
    ProcessorResponseCode["ResponsePpex"] = "PPEX";
    ProcessorResponseCode["ResponsePpfe"] = "PPFE";
    ProcessorResponseCode["ResponsePpfi"] = "PPFI";
    ProcessorResponseCode["ResponsePpfr"] = "PPFR";
    ProcessorResponseCode["ResponsePpfv"] = "PPFV";
    ProcessorResponseCode["ResponsePpgr"] = "PPGR";
    ProcessorResponseCode["ResponsePph1"] = "PPH1";
    ProcessorResponseCode["ResponsePpif"] = "PPIF";
    ProcessorResponseCode["ResponsePpii"] = "PPII";
    ProcessorResponseCode["ResponsePpim"] = "PPIM";
    ProcessorResponseCode["ResponsePpit"] = "PPIT";
    ProcessorResponseCode["ResponsePplr"] = "PPLR";
    ProcessorResponseCode["ResponsePpls"] = "PPLS";
    ProcessorResponseCode["ResponsePpmb"] = "PPMB";
    ProcessorResponseCode["ResponsePpmc"] = "PPMC";
    ProcessorResponseCode["ResponsePpmd"] = "PPMD";
    ProcessorResponseCode["ResponsePpnc"] = "PPNC";
    ProcessorResponseCode["ResponsePpnl"] = "PPNL";
    ProcessorResponseCode["ResponsePpnm"] = "PPNM";
    ProcessorResponseCode["ResponsePpnt"] = "PPNT";
    ProcessorResponseCode["ResponsePpph"] = "PPPH";
    ProcessorResponseCode["ResponsePppi"] = "PPPI";
    ProcessorResponseCode["ResponsePppm"] = "PPPM";
    ProcessorResponseCode["ResponsePpqc"] = "PPQC";
    ProcessorResponseCode["ResponsePpre"] = "PPRE";
    ProcessorResponseCode["ResponsePprf"] = "PPRF";
    ProcessorResponseCode["ResponsePprr"] = "PPRR";
    ProcessorResponseCode["ResponsePps0"] = "PPS0";
    ProcessorResponseCode["ResponsePps1"] = "PPS1";
    ProcessorResponseCode["ResponsePps2"] = "PPS2";
    ProcessorResponseCode["ResponsePps3"] = "PPS3";
    ProcessorResponseCode["ResponsePps4"] = "PPS4";
    ProcessorResponseCode["ResponsePps5"] = "PPS5";
    ProcessorResponseCode["ResponsePps6"] = "PPS6";
    ProcessorResponseCode["ResponsePpsc"] = "PPSC";
    ProcessorResponseCode["ResponsePpsd"] = "PPSD";
    ProcessorResponseCode["ResponsePpse"] = "PPSE";
    ProcessorResponseCode["ResponsePpte"] = "PPTE";
    ProcessorResponseCode["ResponsePptf"] = "PPTF";
    ProcessorResponseCode["ResponsePpti"] = "PPTI";
    ProcessorResponseCode["ResponsePptr"] = "PPTR";
    ProcessorResponseCode["ResponsePptt"] = "PPTT";
    ProcessorResponseCode["ResponsePptv"] = "PPTV";
    ProcessorResponseCode["ResponsePpua"] = "PPUA";
    ProcessorResponseCode["ResponsePpuc"] = "PPUC";
    ProcessorResponseCode["ResponsePpue"] = "PPUE";
    ProcessorResponseCode["ResponsePpui"] = "PPUI";
    ProcessorResponseCode["ResponsePpup"] = "PPUP";
    ProcessorResponseCode["ResponsePpur"] = "PPUR";
    ProcessorResponseCode["ResponsePpvc"] = "PPVC";
    ProcessorResponseCode["ResponsePpve"] = "PPVE";
    ProcessorResponseCode["ResponsePpvt"] = "PPVT";
})(ProcessorResponseCode || (ProcessorResponseCode = {}));
/**
 * Schema for ProcessorResponseCode
 */
export const processorResponseCodeSchema = stringEnum(ProcessorResponseCode, true);
