// @paypal/paypal-server-sdk/dist/esm/models/refund.js
/**
 * PayPal Server SDKLib
 *
 * This file was automatically generated by APIMATIC v3.0 ( https://www.apimatic.io ).
 */
import { array, lazy, object, optional, string } from '../schema';
import { linkDescriptionSchema } from './linkDescription';
import { moneySchema } from './money';
import { payeeBaseSchema } from './payeeBase';
import { refundStatusSchema } from './refundStatus';
import { refundStatusDetailsSchema, } from './refundStatusDetails';
import { sellerPayableBreakdownSchema, } from './sellerPayableBreakdown';
export const refundSchema = object({
    status: ['status', optional(refundStatusSchema)],
    statusDetails: [
        'status_details',
        optional(lazy(() => refundStatusDetailsSchema)),
    ],
    id: ['id', optional(string())],
    amount: ['amount', optional(lazy(() => moneySchema))],
    invoiceId: ['invoice_id', optional(string())],
    customId: ['custom_id', optional(string())],
    acquirerReferenceNumber: ['acquirer_reference_number', optional(string())],
    noteToPayer: ['note_to_payer', optional(string())],
    sellerPayableBreakdown: [
        'seller_payable_breakdown',
        optional(lazy(() => sellerPayableBreakdownSchema)),
    ],
    payer: ['payer', optional(lazy(() => payeeBaseSchema))],
    links: ['links', optional(array(lazy(() => linkDescriptionSchema)))],
    createTime: ['create_time', optional(string())],
    updateTime: ['update_time', optional(string())],
});
