// @paypal/paypal-server-sdk/dist/esm/models/sellerPayableBreakdown.js
/**
 * PayPal Server SDKLib
 *
 * This file was automatically generated by APIMATIC v3.0 ( https://www.apimatic.io ).
 */
import { array, lazy, object, optional } from '../schema';
import { moneySchema } from './money';
import { netAmountBreakdownItemSchema, } from './netAmountBreakdownItem';
import { platformFeeSchema } from './platformFee';
export const sellerPayableBreakdownSchema = object({
    grossAmount: ['gross_amount', optional(lazy(() => moneySchema))],
    paypalFee: ['paypal_fee', optional(lazy(() => moneySchema))],
    paypalFeeInReceivableCurrency: [
        'paypal_fee_in_receivable_currency',
        optional(lazy(() => moneySchema)),
    ],
    netAmount: ['net_amount', optional(lazy(() => moneySchema))],
    netAmountInReceivableCurrency: [
        'net_amount_in_receivable_currency',
        optional(lazy(() => moneySchema)),
    ],
    platformFees: [
        'platform_fees',
        optional(array(lazy(() => platformFeeSchema))),
    ],
    netAmountBreakdown: [
        'net_amount_breakdown',
        optional(array(lazy(() => netAmountBreakdownItemSchema))),
    ],
    totalRefundedAmount: [
        'total_refunded_amount',
        optional(lazy(() => moneySchema)),
    ],
});
