// @paypal/paypal-server-sdk/dist/esm/models/usagePattern.js
/**
 * PayPal Server SDKLib
 *
 * This file was automatically generated by APIMATIC v3.0 ( https://www.apimatic.io ).
 */
import { stringEnum } from '../schema';
/**
 * Enum for UsagePattern
 */
export var UsagePattern;
(function (UsagePattern) {
    UsagePattern["Immediate"] = "IMMEDIATE";
    UsagePattern["Deferred"] = "DEFERRED";
    UsagePattern["RecurringPrepaid"] = "RECURRING_PREPAID";
    UsagePattern["RecurringPostpaid"] = "RECURRING_POSTPAID";
    UsagePattern["ThresholdPrepaid"] = "THRESHOLD_PREPAID";
    UsagePattern["ThresholdPostpaid"] = "THRESHOLD_POSTPAID";
    UsagePattern["SubscriptionPrepaid"] = "SUBSCRIPTION_PREPAID";
    UsagePattern["SubscriptionPostpaid"] = "SUBSCRIPTION_POSTPAID";
    UsagePattern["UnscheduledPrepaid"] = "UNSCHEDULED_PREPAID";
    UsagePattern["UnscheduledPostpaid"] = "UNSCHEDULED_POSTPAID";
    UsagePattern["InstallmentPrepaid"] = "INSTALLMENT_PREPAID";
    UsagePattern["InstallmentPostpaid"] = "INSTALLMENT_POSTPAID";
})(UsagePattern || (UsagePattern = {}));
/**
 * Schema for UsagePattern
 */
export const usagePatternSchema = stringEnum(UsagePattern, true);
