// @paypal/paypal-server-sdk/dist/cjs/models/amountBreakdown.js
"use strict";
/**
 * PayPal Server SDKLib
 *
 * This file was automatically generated by APIMATIC v3.0 ( https://www.apimatic.io ).
 */
exports.__esModule = true;
exports.amountBreakdownSchema = void 0;
var schema_1 = require("../schema");
var money_1 = require("./money");
exports.amountBreakdownSchema = (0, schema_1.object)({
    itemTotal: ['item_total', (0, schema_1.optional)((0, schema_1.lazy)(function () { return money_1.moneySchema; }))],
    shipping: ['shipping', (0, schema_1.optional)((0, schema_1.lazy)(function () { return money_1.moneySchema; }))],
    handling: ['handling', (0, schema_1.optional)((0, schema_1.lazy)(function () { return money_1.moneySchema; }))],
    taxTotal: ['tax_total', (0, schema_1.optional)((0, schema_1.lazy)(function () { return money_1.moneySchema; }))],
    insurance: ['insurance', (0, schema_1.optional)((0, schema_1.lazy)(function () { return money_1.moneySchema; }))],
    shippingDiscount: ['shipping_discount', (0, schema_1.optional)((0, schema_1.lazy)(function () { return money_1.moneySchema; }))],
    discount: ['discount', (0, schema_1.optional)((0, schema_1.lazy)(function () { return money_1.moneySchema; }))]
});
