// @paypal/paypal-server-sdk/dist/cjs/models/applePayCardResponse.js
"use strict";
/**
 * PayPal Server SDKLib
 *
 * This file was automatically generated by APIMATIC v3.0 ( https://www.apimatic.io ).
 */
exports.__esModule = true;
exports.applePayCardResponseSchema = void 0;
var schema_1 = require("../schema");
var address_1 = require("./address");
var authenticationResponse_1 = require("./authenticationResponse");
var binDetails_1 = require("./binDetails");
var cardAttributesResponse_1 = require("./cardAttributesResponse");
var cardBrand_1 = require("./cardBrand");
var cardFromRequest_1 = require("./cardFromRequest");
var cardStoredCredential_1 = require("./cardStoredCredential");
var cardType_1 = require("./cardType");
exports.applePayCardResponseSchema = (0, schema_1.object)({
    name: ['name', (0, schema_1.optional)((0, schema_1.string)())],
    lastDigits: ['last_digits', (0, schema_1.optional)((0, schema_1.string)())],
    brand: ['brand', (0, schema_1.optional)(cardBrand_1.cardBrandSchema)],
    availableNetworks: ['available_networks', (0, schema_1.optional)((0, schema_1.array)(cardBrand_1.cardBrandSchema))],
    type: ['type', (0, schema_1.optional)(cardType_1.cardTypeSchema)],
    authenticationResult: [
        'authentication_result',
        (0, schema_1.optional)((0, schema_1.lazy)(function () { return authenticationResponse_1.authenticationResponseSchema; })),
    ],
    attributes: [
        'attributes',
        (0, schema_1.optional)((0, schema_1.lazy)(function () { return cardAttributesResponse_1.cardAttributesResponseSchema; })),
    ],
    fromRequest: ['from_request', (0, schema_1.optional)((0, schema_1.lazy)(function () { return cardFromRequest_1.cardFromRequestSchema; }))],
    expiry: ['expiry', (0, schema_1.optional)((0, schema_1.string)())],
    binDetails: ['bin_details', (0, schema_1.optional)((0, schema_1.lazy)(function () { return binDetails_1.binDetailsSchema; }))],
    storedCredential: [
        'stored_credential',
        (0, schema_1.optional)((0, schema_1.lazy)(function () { return cardStoredCredential_1.cardStoredCredentialSchema; })),
    ],
    billingAddress: ['billing_address', (0, schema_1.optional)((0, schema_1.lazy)(function () { return address_1.addressSchema; }))],
    countryCode: ['country_code', (0, schema_1.optional)((0, schema_1.string)())]
});
