// @paypal/paypal-server-sdk/dist/cjs/models/applePayDecryptedTokenData.js
"use strict";
/**
 * PayPal Server SDKLib
 *
 * This file was automatically generated by APIMATIC v3.0 ( https://www.apimatic.io ).
 */
exports.__esModule = true;
exports.applePayDecryptedTokenDataSchema = void 0;
var schema_1 = require("../schema");
var applePayPaymentData_1 = require("./applePayPaymentData");
var applePayPaymentDataType_1 = require("./applePayPaymentDataType");
var applePayTokenizedCard_1 = require("./applePayTokenizedCard");
var money_1 = require("./money");
exports.applePayDecryptedTokenDataSchema = (0, schema_1.object)({
    transactionAmount: [
        'transaction_amount',
        (0, schema_1.optional)((0, schema_1.lazy)(function () { return money_1.moneySchema; })),
    ],
    tokenizedCard: ['tokenized_card', (0, schema_1.lazy)(function () { return applePayTokenizedCard_1.applePayTokenizedCardSchema; })],
    deviceManufacturerId: ['device_manufacturer_id', (0, schema_1.optional)((0, schema_1.string)())],
    paymentDataType: [
        'payment_data_type',
        (0, schema_1.optional)(applePayPaymentDataType_1.applePayPaymentDataTypeSchema),
    ],
    paymentData: [
        'payment_data',
        (0, schema_1.optional)((0, schema_1.lazy)(function () { return applePayPaymentData_1.applePayPaymentDataSchema; })),
    ]
});
