// @paypal/paypal-server-sdk/dist/cjs/models/applePayTokenizedCard.js
"use strict";
/**
 * PayPal Server SDKLib
 *
 * This file was automatically generated by APIMATIC v3.0 ( https://www.apimatic.io ).
 */
exports.__esModule = true;
exports.applePayTokenizedCardSchema = void 0;
var schema_1 = require("../schema");
var address_1 = require("./address");
var cardBrand_1 = require("./cardBrand");
var cardType_1 = require("./cardType");
exports.applePayTokenizedCardSchema = (0, schema_1.object)({
    name: ['name', (0, schema_1.optional)((0, schema_1.string)())],
    number: ['number', (0, schema_1.optional)((0, schema_1.string)())],
    expiry: ['expiry', (0, schema_1.optional)((0, schema_1.string)())],
    cardType: ['card_type', (0, schema_1.optional)(cardBrand_1.cardBrandSchema)],
    type: ['type', (0, schema_1.optional)(cardType_1.cardTypeSchema)],
    brand: ['brand', (0, schema_1.optional)(cardBrand_1.cardBrandSchema)],
    billingAddress: ['billing_address', (0, schema_1.optional)((0, schema_1.lazy)(function () { return address_1.addressSchema; }))]
});
