// @paypal/paypal-server-sdk/dist/cjs/models/authorization.js
"use strict";
/**
 * PayPal Server SDKLib
 *
 * This file was automatically generated by APIMATIC v3.0 ( https://www.apimatic.io ).
 */
exports.__esModule = true;
exports.authorizationSchema = void 0;
var schema_1 = require("../schema");
var authorizationStatus_1 = require("./authorizationStatus");
var authorizationStatusDetails_1 = require("./authorizationStatusDetails");
var linkDescription_1 = require("./linkDescription");
var money_1 = require("./money");
var networkTransactionReference_1 = require("./networkTransactionReference");
var sellerProtection_1 = require("./sellerProtection");
exports.authorizationSchema = (0, schema_1.object)({
    status: ['status', (0, schema_1.optional)(authorizationStatus_1.authorizationStatusSchema)],
    statusDetails: [
        'status_details',
        (0, schema_1.optional)((0, schema_1.lazy)(function () { return authorizationStatusDetails_1.authorizationStatusDetailsSchema; })),
    ],
    id: ['id', (0, schema_1.optional)((0, schema_1.string)())],
    amount: ['amount', (0, schema_1.optional)((0, schema_1.lazy)(function () { return money_1.moneySchema; }))],
    invoiceId: ['invoice_id', (0, schema_1.optional)((0, schema_1.string)())],
    customId: ['custom_id', (0, schema_1.optional)((0, schema_1.string)())],
    networkTransactionReference: [
        'network_transaction_reference',
        (0, schema_1.optional)((0, schema_1.lazy)(function () { return networkTransactionReference_1.networkTransactionReferenceSchema; })),
    ],
    sellerProtection: [
        'seller_protection',
        (0, schema_1.optional)((0, schema_1.lazy)(function () { return sellerProtection_1.sellerProtectionSchema; })),
    ],
    expirationTime: ['expiration_time', (0, schema_1.optional)((0, schema_1.string)())],
    links: ['links', (0, schema_1.optional)((0, schema_1.array)((0, schema_1.lazy)(function () { return linkDescription_1.linkDescriptionSchema; })))],
    createTime: ['create_time', (0, schema_1.optional)((0, schema_1.string)())],
    updateTime: ['update_time', (0, schema_1.optional)((0, schema_1.string)())]
});
