// @paypal/paypal-server-sdk/dist/cjs/models/avsCode.js
"use strict";
/**
 * PayPal Server SDKLib
 *
 * This file was automatically generated by APIMATIC v3.0 ( https://www.apimatic.io ).
 */
exports.__esModule = true;
exports.avsCodeSchema = exports.AvsCode = void 0;
var schema_1 = require("../schema");
/**
 * Enum for AvsCode
 */
var AvsCode;
(function (AvsCode) {
    AvsCode["AvsA"] = "A";
    AvsCode["AvsB"] = "B";
    AvsCode["AvsC"] = "C";
    AvsCode["AvsD"] = "D";
    AvsCode["AvsE"] = "E";
    AvsCode["AvsF"] = "F";
    AvsCode["AvsG"] = "G";
    AvsCode["AvsI"] = "I";
    AvsCode["AvsM"] = "M";
    AvsCode["AvsN"] = "N";
    AvsCode["AvsP"] = "P";
    AvsCode["AvsR"] = "R";
    AvsCode["AvsS"] = "S";
    AvsCode["AvsU"] = "U";
    AvsCode["AvsW"] = "W";
    AvsCode["AvsX"] = "X";
    AvsCode["AvsY"] = "Y";
    AvsCode["AvsZ"] = "Z";
    AvsCode["AvsNull"] = "Null";
    AvsCode["Avs0"] = "0";
    AvsCode["Avs1"] = "1";
    AvsCode["Avs2"] = "2";
    AvsCode["Avs3"] = "3";
    AvsCode["Avs4"] = "4";
})(AvsCode = exports.AvsCode || (exports.AvsCode = {}));
/**
 * Schema for AvsCode
 */
exports.avsCodeSchema = (0, schema_1.stringEnum)(AvsCode, true);
