// @paypal/paypal-server-sdk/dist/cjs/models/blikPaymentRequest.js
"use strict";
/**
 * PayPal Server SDKLib
 *
 * This file was automatically generated by APIMATIC v3.0 ( https://www.apimatic.io ).
 */
exports.__esModule = true;
exports.blikPaymentRequestSchema = void 0;
var schema_1 = require("../schema");
var blikExperienceContext_1 = require("./blikExperienceContext");
var blikLevel0PaymentObject_1 = require("./blikLevel0PaymentObject");
var blikOneClickPaymentRequest_1 = require("./blikOneClickPaymentRequest");
exports.blikPaymentRequestSchema = (0, schema_1.object)({
    name: ['name', (0, schema_1.string)()],
    countryCode: ['country_code', (0, schema_1.string)()],
    email: ['email', (0, schema_1.optional)((0, schema_1.string)())],
    experienceContext: [
        'experience_context',
        (0, schema_1.optional)((0, schema_1.lazy)(function () { return blikExperienceContext_1.blikExperienceContextSchema; })),
    ],
    level0: ['level_0', (0, schema_1.optional)((0, schema_1.lazy)(function () { return blikLevel0PaymentObject_1.blikLevel0PaymentObjectSchema; }))],
    oneClick: [
        'one_click',
        (0, schema_1.optional)((0, schema_1.lazy)(function () { return blikOneClickPaymentRequest_1.blikOneClickPaymentRequestSchema; })),
    ]
});
