// @paypal/paypal-server-sdk/dist/cjs/models/captureIncompleteReason.js
"use strict";
/**
 * PayPal Server SDKLib
 *
 * This file was automatically generated by APIMATIC v3.0 ( https://www.apimatic.io ).
 */
exports.__esModule = true;
exports.captureIncompleteReasonSchema = exports.CaptureIncompleteReason = void 0;
var schema_1 = require("../schema");
/**
 * Enum for CaptureIncompleteReason
 */
var CaptureIncompleteReason;
(function (CaptureIncompleteReason) {
    CaptureIncompleteReason["BuyerComplaint"] = "BUYER_COMPLAINT";
    CaptureIncompleteReason["Chargeback"] = "CHARGEBACK";
    CaptureIncompleteReason["Echeck"] = "ECHECK";
    CaptureIncompleteReason["InternationalWithdrawal"] = "INTERNATIONAL_WITHDRAWAL";
    CaptureIncompleteReason["Other"] = "OTHER";
    CaptureIncompleteReason["PendingReview"] = "PENDING_REVIEW";
    CaptureIncompleteReason["ReceivingPreferenceMandatesManualAction"] = "RECEIVING_PREFERENCE_MANDATES_MANUAL_ACTION";
    CaptureIncompleteReason["Refunded"] = "REFUNDED";
    CaptureIncompleteReason["TransactionApprovedAwaitingFunding"] = "TRANSACTION_APPROVED_AWAITING_FUNDING";
    CaptureIncompleteReason["Unilateral"] = "UNILATERAL";
    CaptureIncompleteReason["VerificationRequired"] = "VERIFICATION_REQUIRED";
    CaptureIncompleteReason["DeclinedByRiskFraudFilters"] = "DECLINED_BY_RISK_FRAUD_FILTERS";
})(CaptureIncompleteReason = exports.CaptureIncompleteReason || (exports.CaptureIncompleteReason = {}));
/**
 * Schema for CaptureIncompleteReason
 */
exports.captureIncompleteReasonSchema = (0, schema_1.stringEnum)(CaptureIncompleteReason, true);
