// @paypal/paypal-server-sdk/dist/cjs/models/captureRequest.js
"use strict";
/**
 * PayPal Server SDKLib
 *
 * This file was automatically generated by APIMATIC v3.0 ( https://www.apimatic.io ).
 */
exports.__esModule = true;
exports.captureRequestSchema = void 0;
var schema_1 = require("../schema");
var capturePaymentInstruction_1 = require("./capturePaymentInstruction");
var money_1 = require("./money");
exports.captureRequestSchema = (0, schema_1.object)({
    invoiceId: ['invoice_id', (0, schema_1.optional)((0, schema_1.string)())],
    noteToPayer: ['note_to_payer', (0, schema_1.optional)((0, schema_1.string)())],
    amount: ['amount', (0, schema_1.optional)((0, schema_1.lazy)(function () { return money_1.moneySchema; }))],
    finalCapture: ['final_capture', (0, schema_1.optional)((0, schema_1.boolean)())],
    paymentInstruction: [
        'payment_instruction',
        (0, schema_1.optional)((0, schema_1.lazy)(function () { return capturePaymentInstruction_1.capturePaymentInstructionSchema; })),
    ],
    softDescriptor: ['soft_descriptor', (0, schema_1.optional)((0, schema_1.string)())]
});
