// @paypal/paypal-server-sdk/dist/cjs/models/cardVerificationDetails.js
"use strict";
/**
 * PayPal Server SDKLib
 *
 * This file was automatically generated by APIMATIC v3.0 ( https://www.apimatic.io ).
 */
exports.__esModule = true;
exports.cardVerificationDetailsSchema = void 0;
var schema_1 = require("../schema");
var cardBrand_1 = require("./cardBrand");
var cardVerificationProcessorResponse_1 = require("./cardVerificationProcessorResponse");
var money_1 = require("./money");
exports.cardVerificationDetailsSchema = (0, schema_1.object)({
    networkTransactionId: ['network_transaction_id', (0, schema_1.optional)((0, schema_1.string)())],
    date: ['date', (0, schema_1.optional)((0, schema_1.string)())],
    network: ['network', (0, schema_1.optional)(cardBrand_1.cardBrandSchema)],
    time: ['time', (0, schema_1.optional)((0, schema_1.string)())],
    amount: ['amount', (0, schema_1.optional)((0, schema_1.lazy)(function () { return money_1.moneySchema; }))],
    processorResponse: [
        'processor_response',
        (0, schema_1.optional)((0, schema_1.lazy)(function () { return cardVerificationProcessorResponse_1.cardVerificationProcessorResponseSchema; })),
    ],
    threeDSecure: ['three_d_secure', (0, schema_1.optional)((0, schema_1.unknown)())]
});
