// @paypal/paypal-server-sdk/dist/cjs/models/customerVaultPaymentTokensResponse.js
"use strict";
/**
 * PayPal Server SDKLib
 *
 * This file was automatically generated by APIMATIC v3.0 ( https://www.apimatic.io ).
 */
exports.__esModule = true;
exports.customerVaultPaymentTokensResponseSchema = void 0;
var schema_1 = require("../schema");
var linkDescription_1 = require("./linkDescription");
var paymentTokenResponse_1 = require("./paymentTokenResponse");
var vaultResponseCustomer_1 = require("./vaultResponseCustomer");
exports.customerVaultPaymentTokensResponseSchema = (0, schema_1.object)({
    totalItems: ['total_items', (0, schema_1.optional)((0, schema_1.number)())],
    totalPages: ['total_pages', (0, schema_1.optional)((0, schema_1.number)())],
    customer: ['customer', (0, schema_1.optional)((0, schema_1.lazy)(function () { return vaultResponseCustomer_1.vaultResponseCustomerSchema; }))],
    paymentTokens: [
        'payment_tokens',
        (0, schema_1.optional)((0, schema_1.array)((0, schema_1.lazy)(function () { return paymentTokenResponse_1.paymentTokenResponseSchema; }))),
    ],
    links: ['links', (0, schema_1.optional)((0, schema_1.array)((0, schema_1.lazy)(function () { return linkDescription_1.linkDescriptionSchema; })))]
});
