// @paypal/paypal-server-sdk/dist/cjs/models/eciFlag.js
"use strict";
/**
 * PayPal Server SDKLib
 *
 * This file was automatically generated by APIMATIC v3.0 ( https://www.apimatic.io ).
 */
exports.__esModule = true;
exports.eciFlagSchema = exports.EciFlag = void 0;
var schema_1 = require("../schema");
/**
 * Enum for EciFlag
 */
var EciFlag;
(function (EciFlag) {
    EciFlag["MastercardNon3DSecureTransaction"] = "MASTERCARD_NON_3D_SECURE_TRANSACTION";
    EciFlag["MastercardAttemptedAuthenticationTransaction"] = "MASTERCARD_ATTEMPTED_AUTHENTICATION_TRANSACTION";
    EciFlag["MastercardFullyAuthenticatedTransaction"] = "MASTERCARD_FULLY_AUTHENTICATED_TRANSACTION";
    EciFlag["FullyAuthenticatedTransaction"] = "FULLY_AUTHENTICATED_TRANSACTION";
    EciFlag["AttemptedAuthenticationTransaction"] = "ATTEMPTED_AUTHENTICATION_TRANSACTION";
    EciFlag["Non3DSecureTransaction"] = "NON_3D_SECURE_TRANSACTION";
})(EciFlag = exports.EciFlag || (exports.EciFlag = {}));
/**
 * Schema for EciFlag
 */
exports.eciFlagSchema = (0, schema_1.stringEnum)(EciFlag, true);
