// @paypal/paypal-server-sdk/dist/cjs/models/googlePayCardResponse.js
"use strict";
/**
 * PayPal Server SDKLib
 *
 * This file was automatically generated by APIMATIC v3.0 ( https://www.apimatic.io ).
 */
exports.__esModule = true;
exports.googlePayCardResponseSchema = void 0;
var schema_1 = require("../schema");
var address_1 = require("./address");
var authenticationResponse_1 = require("./authenticationResponse");
var cardBrand_1 = require("./cardBrand");
var cardType_1 = require("./cardType");
exports.googlePayCardResponseSchema = (0, schema_1.object)({
    name: ['name', (0, schema_1.optional)((0, schema_1.string)())],
    lastDigits: ['last_digits', (0, schema_1.optional)((0, schema_1.string)())],
    type: ['type', (0, schema_1.optional)(cardType_1.cardTypeSchema)],
    brand: ['brand', (0, schema_1.optional)(cardBrand_1.cardBrandSchema)],
    billingAddress: ['billing_address', (0, schema_1.optional)((0, schema_1.lazy)(function () { return address_1.addressSchema; }))],
    authenticationResult: [
        'authentication_result',
        (0, schema_1.optional)((0, schema_1.lazy)(function () { return authenticationResponse_1.authenticationResponseSchema; })),
    ]
});
