// @paypal/paypal-server-sdk/dist/cjs/models/googlePayDecryptedTokenData.js
"use strict";
/**
 * PayPal Server SDKLib
 *
 * This file was automatically generated by APIMATIC v3.0 ( https://www.apimatic.io ).
 */
exports.__esModule = true;
exports.googlePayDecryptedTokenDataSchema = void 0;
var schema_1 = require("../schema");
var googlePayAuthenticationMethod_1 = require("./googlePayAuthenticationMethod");
var googlePayCard_1 = require("./googlePayCard");
var googlePayPaymentMethod_1 = require("./googlePayPaymentMethod");
exports.googlePayDecryptedTokenDataSchema = (0, schema_1.object)({
    messageId: ['message_id', (0, schema_1.optional)((0, schema_1.string)())],
    messageExpiration: ['message_expiration', (0, schema_1.optional)((0, schema_1.string)())],
    paymentMethod: ['payment_method', googlePayPaymentMethod_1.googlePayPaymentMethodSchema],
    card: ['card', (0, schema_1.lazy)(function () { return googlePayCard_1.googlePayCardSchema; })],
    authenticationMethod: [
        'authentication_method',
        googlePayAuthenticationMethod_1.googlePayAuthenticationMethodSchema,
    ],
    cryptogram: ['cryptogram', (0, schema_1.optional)((0, schema_1.string)())],
    eciIndicator: ['eci_indicator', (0, schema_1.optional)((0, schema_1.string)())]
});
