// @paypal/paypal-server-sdk/dist/cjs/models/item.js
"use strict";
/**
 * PayPal Server SDKLib
 *
 * This file was automatically generated by APIMATIC v3.0 ( https://www.apimatic.io ).
 */
exports.__esModule = true;
exports.itemSchema = void 0;
var schema_1 = require("../schema");
var itemCategory_1 = require("./itemCategory");
var money_1 = require("./money");
var orderBillingPlan_1 = require("./orderBillingPlan");
var universalProductCode_1 = require("./universalProductCode");
exports.itemSchema = (0, schema_1.object)({
    name: ['name', (0, schema_1.string)()],
    unitAmount: ['unit_amount', (0, schema_1.lazy)(function () { return money_1.moneySchema; })],
    tax: ['tax', (0, schema_1.optional)((0, schema_1.lazy)(function () { return money_1.moneySchema; }))],
    quantity: ['quantity', (0, schema_1.string)()],
    description: ['description', (0, schema_1.optional)((0, schema_1.string)())],
    sku: ['sku', (0, schema_1.optional)((0, schema_1.string)())],
    url: ['url', (0, schema_1.optional)((0, schema_1.string)())],
    category: ['category', (0, schema_1.optional)(itemCategory_1.itemCategorySchema)],
    imageUrl: ['image_url', (0, schema_1.optional)((0, schema_1.string)())],
    upc: ['upc', (0, schema_1.optional)((0, schema_1.lazy)(function () { return universalProductCode_1.universalProductCodeSchema; }))],
    billingPlan: ['billing_plan', (0, schema_1.optional)((0, schema_1.lazy)(function () { return orderBillingPlan_1.orderBillingPlanSchema; }))]
});
