// @paypal/paypal-server-sdk/dist/cjs/models/oneTimeCharge.js
"use strict";
/**
 * PayPal Server SDKLib
 *
 * This file was automatically generated by APIMATIC v3.0 ( https://www.apimatic.io ).
 */
exports.__esModule = true;
exports.oneTimeChargeSchema = void 0;
var schema_1 = require("../schema");
var money_1 = require("./money");
exports.oneTimeChargeSchema = (0, schema_1.object)({
    setupFee: ['setup_fee', (0, schema_1.optional)((0, schema_1.lazy)(function () { return money_1.moneySchema; }))],
    shippingAmount: ['shipping_amount', (0, schema_1.optional)((0, schema_1.lazy)(function () { return money_1.moneySchema; }))],
    taxes: ['taxes', (0, schema_1.optional)((0, schema_1.lazy)(function () { return money_1.moneySchema; }))],
    productPrice: ['product_price', (0, schema_1.optional)((0, schema_1.lazy)(function () { return money_1.moneySchema; }))],
    subtotal: ['subtotal', (0, schema_1.optional)((0, schema_1.lazy)(function () { return money_1.moneySchema; }))],
    totalAmount: ['total_amount', (0, schema_1.lazy)(function () { return money_1.moneySchema; })]
});
