// @paypal/paypal-server-sdk/dist/cjs/models/orderCaptureRequestPaymentSource.js
"use strict";
/**
 * PayPal Server SDKLib
 *
 * This file was automatically generated by APIMATIC v3.0 ( https://www.apimatic.io ).
 */
exports.__esModule = true;
exports.orderCaptureRequestPaymentSourceSchema = void 0;
var schema_1 = require("../schema");
var applePayRequest_1 = require("./applePayRequest");
var cardRequest_1 = require("./cardRequest");
var googlePayRequest_1 = require("./googlePayRequest");
var paypalWallet_1 = require("./paypalWallet");
var token_1 = require("./token");
var venmoWalletRequest_1 = require("./venmoWalletRequest");
exports.orderCaptureRequestPaymentSourceSchema = (0, schema_1.object)({
    card: ['card', (0, schema_1.optional)((0, schema_1.lazy)(function () { return cardRequest_1.cardRequestSchema; }))],
    token: ['token', (0, schema_1.optional)((0, schema_1.lazy)(function () { return token_1.tokenSchema; }))],
    paypal: ['paypal', (0, schema_1.optional)((0, schema_1.lazy)(function () { return paypalWallet_1.paypalWalletSchema; }))],
    applePay: ['apple_pay', (0, schema_1.optional)((0, schema_1.lazy)(function () { return applePayRequest_1.applePayRequestSchema; }))],
    googlePay: ['google_pay', (0, schema_1.optional)((0, schema_1.lazy)(function () { return googlePayRequest_1.googlePayRequestSchema; }))],
    venmo: ['venmo', (0, schema_1.optional)((0, schema_1.lazy)(function () { return venmoWalletRequest_1.venmoWalletRequestSchema; }))]
});
