// @paypal/paypal-server-sdk/dist/cjs/models/orderApplicationContext.js
"use strict";
/**
 * PayPal Server SDKLib
 *
 * This file was automatically generated by APIMATIC v3.0 ( https://www.apimatic.io ).
 */
exports.__esModule = true;
exports.orderApplicationContextSchema = void 0;
var schema_1 = require("../schema");
var orderApplicationContextLandingPage_1 = require("./orderApplicationContextLandingPage");
var orderApplicationContextShippingPreference_1 = require("./orderApplicationContextShippingPreference");
var orderApplicationContextUserAction_1 = require("./orderApplicationContextUserAction");
var paymentMethodPreference_1 = require("./paymentMethodPreference");
var storedPaymentSource_1 = require("./storedPaymentSource");
exports.orderApplicationContextSchema = (0, schema_1.object)({
    brandName: ['brand_name', (0, schema_1.optional)((0, schema_1.string)())],
    locale: ['locale', (0, schema_1.optional)((0, schema_1.string)())],
    landingPage: [
        'landing_page',
        (0, schema_1.optional)(orderApplicationContextLandingPage_1.orderApplicationContextLandingPageSchema),
    ],
    shippingPreference: [
        'shipping_preference',
        (0, schema_1.optional)(orderApplicationContextShippingPreference_1.orderApplicationContextShippingPreferenceSchema),
    ],
    userAction: [
        'user_action',
        (0, schema_1.optional)(orderApplicationContextUserAction_1.orderApplicationContextUserActionSchema),
    ],
    paymentMethod: [
        'payment_method',
        (0, schema_1.optional)((0, schema_1.lazy)(function () { return paymentMethodPreference_1.paymentMethodPreferenceSchema; })),
    ],
    returnUrl: ['return_url', (0, schema_1.optional)((0, schema_1.string)())],
    cancelUrl: ['cancel_url', (0, schema_1.optional)((0, schema_1.string)())],
    storedPaymentSource: [
        'stored_payment_source',
        (0, schema_1.optional)((0, schema_1.lazy)(function () { return storedPaymentSource_1.storedPaymentSourceSchema; })),
    ]
});
