// @paypal/paypal-server-sdk/dist/cjs/models/orderTrackerResponse.js
"use strict";
/**
 * PayPal Server SDKLib
 *
 * This file was automatically generated by APIMATIC v3.0 ( https://www.apimatic.io ).
 */
exports.__esModule = true;
exports.orderTrackerResponseSchema = void 0;
var schema_1 = require("../schema");
var linkDescription_1 = require("./linkDescription");
var orderTrackerItem_1 = require("./orderTrackerItem");
var orderTrackerStatus_1 = require("./orderTrackerStatus");
exports.orderTrackerResponseSchema = (0, schema_1.object)({
    id: ['id', (0, schema_1.optional)((0, schema_1.string)())],
    status: ['status', (0, schema_1.optional)(orderTrackerStatus_1.orderTrackerStatusSchema)],
    items: ['items', (0, schema_1.optional)((0, schema_1.array)((0, schema_1.lazy)(function () { return orderTrackerItem_1.orderTrackerItemSchema; })))],
    links: ['links', (0, schema_1.optional)((0, schema_1.array)((0, schema_1.lazy)(function () { return linkDescription_1.linkDescriptionSchema; })))],
    createTime: ['create_time', (0, schema_1.optional)((0, schema_1.string)())],
    updateTime: ['update_time', (0, schema_1.optional)((0, schema_1.string)())]
});
