// @paypal/paypal-server-sdk/dist/cjs/models/orderAuthorizeResponsePaymentSource.js
"use strict";
/**
 * PayPal Server SDKLib
 *
 * This file was automatically generated by APIMATIC v3.0 ( https://www.apimatic.io ).
 */
exports.__esModule = true;
exports.orderAuthorizeResponsePaymentSourceSchema = void 0;
var schema_1 = require("../schema");
var applePayPaymentObject_1 = require("./applePayPaymentObject");
var cardResponse_1 = require("./cardResponse");
var googlePayWalletResponse_1 = require("./googlePayWalletResponse");
var paypalWalletResponse_1 = require("./paypalWalletResponse");
var venmoWalletResponse_1 = require("./venmoWalletResponse");
exports.orderAuthorizeResponsePaymentSourceSchema = (0, schema_1.object)({
    card: ['card', (0, schema_1.optional)((0, schema_1.lazy)(function () { return cardResponse_1.cardResponseSchema; }))],
    paypal: ['paypal', (0, schema_1.optional)((0, schema_1.lazy)(function () { return paypalWalletResponse_1.paypalWalletResponseSchema; }))],
    applePay: ['apple_pay', (0, schema_1.optional)((0, schema_1.lazy)(function () { return applePayPaymentObject_1.applePayPaymentObjectSchema; }))],
    googlePay: [
        'google_pay',
        (0, schema_1.optional)((0, schema_1.lazy)(function () { return googlePayWalletResponse_1.googlePayWalletResponseSchema; })),
    ],
    venmo: ['venmo', (0, schema_1.optional)((0, schema_1.lazy)(function () { return venmoWalletResponse_1.venmoWalletResponseSchema; }))]
});
