// @paypal/paypal-server-sdk/dist/cjs/models/orderAuthorizeResponse.js
"use strict";
/**
 * PayPal Server SDKLib
 *
 * This file was automatically generated by APIMATIC v3.0 ( https://www.apimatic.io ).
 */
exports.__esModule = true;
exports.orderAuthorizeResponseSchema = void 0;
var schema_1 = require("../schema");
var checkoutPaymentIntent_1 = require("./checkoutPaymentIntent");
var linkDescription_1 = require("./linkDescription");
var orderAuthorizeResponsePaymentSource_1 = require("./orderAuthorizeResponsePaymentSource");
var orderStatus_1 = require("./orderStatus");
var payer_1 = require("./payer");
var purchaseUnit_1 = require("./purchaseUnit");
exports.orderAuthorizeResponseSchema = (0, schema_1.object)({
    createTime: ['create_time', (0, schema_1.optional)((0, schema_1.string)())],
    updateTime: ['update_time', (0, schema_1.optional)((0, schema_1.string)())],
    id: ['id', (0, schema_1.optional)((0, schema_1.string)())],
    paymentSource: [
        'payment_source',
        (0, schema_1.optional)((0, schema_1.lazy)(function () { return orderAuthorizeResponsePaymentSource_1.orderAuthorizeResponsePaymentSourceSchema; })),
    ],
    intent: ['intent', (0, schema_1.optional)(checkoutPaymentIntent_1.checkoutPaymentIntentSchema)],
    payer: ['payer', (0, schema_1.optional)((0, schema_1.lazy)(function () { return payer_1.payerSchema; }))],
    purchaseUnits: [
        'purchase_units',
        (0, schema_1.optional)((0, schema_1.array)((0, schema_1.lazy)(function () { return purchaseUnit_1.purchaseUnitSchema; }))),
    ],
    status: ['status', (0, schema_1.optional)(orderStatus_1.orderStatusSchema)],
    links: ['links', (0, schema_1.optional)((0, schema_1.array)((0, schema_1.lazy)(function () { return linkDescription_1.linkDescriptionSchema; })))]
});
