// @paypal/paypal-server-sdk/dist/cjs/models/orderRequest.js
"use strict";
/**
 * PayPal Server SDKLib
 *
 * This file was automatically generated by APIMATIC v3.0 ( https://www.apimatic.io ).
 */
exports.__esModule = true;
exports.orderRequestSchema = void 0;
var schema_1 = require("../schema");
var checkoutPaymentIntent_1 = require("./checkoutPaymentIntent");
var orderApplicationContext_1 = require("./orderApplicationContext");
var payer_1 = require("./payer");
var paymentSource_1 = require("./paymentSource");
var purchaseUnitRequest_1 = require("./purchaseUnitRequest");
exports.orderRequestSchema = (0, schema_1.object)({
    intent: ['intent', checkoutPaymentIntent_1.checkoutPaymentIntentSchema],
    payer: ['payer', (0, schema_1.optional)((0, schema_1.lazy)(function () { return payer_1.payerSchema; }))],
    purchaseUnits: [
        'purchase_units',
        (0, schema_1.array)((0, schema_1.lazy)(function () { return purchaseUnitRequest_1.purchaseUnitRequestSchema; })),
    ],
    paymentSource: ['payment_source', (0, schema_1.optional)((0, schema_1.lazy)(function () { return paymentSource_1.paymentSourceSchema; }))],
    applicationContext: [
        'application_context',
        (0, schema_1.optional)((0, schema_1.lazy)(function () { return orderApplicationContext_1.orderApplicationContextSchema; })),
    ]
});
