// @paypal/paypal-server-sdk/dist/cjs/models/payer.js
"use strict";
/**
 * PayPal Server SDKLib
 *
 * This file was automatically generated by APIMATIC v3.0 ( https://www.apimatic.io ).
 */
exports.__esModule = true;
exports.payerSchema = void 0;
var schema_1 = require("../schema");
var address_1 = require("./address");
var name_1 = require("./name");
var phoneWithType_1 = require("./phoneWithType");
var taxInfo_1 = require("./taxInfo");
exports.payerSchema = (0, schema_1.object)({
    emailAddress: ['email_address', (0, schema_1.optional)((0, schema_1.string)())],
    payerId: ['payer_id', (0, schema_1.optional)((0, schema_1.string)())],
    name: ['name', (0, schema_1.optional)((0, schema_1.lazy)(function () { return name_1.nameSchema; }))],
    phone: ['phone', (0, schema_1.optional)((0, schema_1.lazy)(function () { return phoneWithType_1.phoneWithTypeSchema; }))],
    birthDate: ['birth_date', (0, schema_1.optional)((0, schema_1.string)())],
    taxInfo: ['tax_info', (0, schema_1.optional)((0, schema_1.lazy)(function () { return taxInfo_1.taxInfoSchema; }))],
    address: ['address', (0, schema_1.optional)((0, schema_1.lazy)(function () { return address_1.addressSchema; }))]
});
