// @paypal/paypal-server-sdk/dist/cjs/models/paymentsCapture.js
"use strict";
/**
 * PayPal Server SDKLib
 *
 * This file was automatically generated by APIMATIC v3.0 ( https://www.apimatic.io ).
 */
exports.__esModule = true;
exports.paymentsCaptureSchema = void 0;
var schema_1 = require("../schema");
var captureStatus_1 = require("./captureStatus");
var captureStatusDetails_1 = require("./captureStatusDetails");
var disbursementMode_1 = require("./disbursementMode");
var linkDescription_1 = require("./linkDescription");
var money_1 = require("./money");
var networkTransactionReference_1 = require("./networkTransactionReference");
var paymentsProcessorResponse_1 = require("./paymentsProcessorResponse");
var sellerProtection_1 = require("./sellerProtection");
var sellerReceivableBreakdown_1 = require("./sellerReceivableBreakdown");
exports.paymentsCaptureSchema = (0, schema_1.object)({
    status: ['status', (0, schema_1.optional)(captureStatus_1.captureStatusSchema)],
    statusDetails: [
        'status_details',
        (0, schema_1.optional)((0, schema_1.lazy)(function () { return captureStatusDetails_1.captureStatusDetailsSchema; })),
    ],
    id: ['id', (0, schema_1.optional)((0, schema_1.string)())],
    amount: ['amount', (0, schema_1.optional)((0, schema_1.lazy)(function () { return money_1.moneySchema; }))],
    invoiceId: ['invoice_id', (0, schema_1.optional)((0, schema_1.string)())],
    customId: ['custom_id', (0, schema_1.optional)((0, schema_1.string)())],
    networkTransactionReference: [
        'network_transaction_reference',
        (0, schema_1.optional)((0, schema_1.lazy)(function () { return networkTransactionReference_1.networkTransactionReferenceSchema; })),
    ],
    sellerProtection: [
        'seller_protection',
        (0, schema_1.optional)((0, schema_1.lazy)(function () { return sellerProtection_1.sellerProtectionSchema; })),
    ],
    finalCapture: ['final_capture', (0, schema_1.optional)((0, schema_1.boolean)())],
    sellerReceivableBreakdown: [
        'seller_receivable_breakdown',
        (0, schema_1.optional)((0, schema_1.lazy)(function () { return sellerReceivableBreakdown_1.sellerReceivableBreakdownSchema; })),
    ],
    disbursementMode: ['disbursement_mode', (0, schema_1.optional)(disbursementMode_1.disbursementModeSchema)],
    links: ['links', (0, schema_1.optional)((0, schema_1.array)((0, schema_1.lazy)(function () { return linkDescription_1.linkDescriptionSchema; })))],
    processorResponse: [
        'processor_response',
        (0, schema_1.optional)((0, schema_1.lazy)(function () { return paymentsProcessorResponse_1.paymentsProcessorResponseSchema; })),
    ],
    createTime: ['create_time', (0, schema_1.optional)((0, schema_1.string)())],
    updateTime: ['update_time', (0, schema_1.optional)((0, schema_1.string)())]
});
