// @paypal/paypal-server-sdk/dist/cjs/models/purchaseUnit.js
"use strict";
/**
 * PayPal Server SDKLib
 *
 * This file was automatically generated by APIMATIC v3.0 ( https://www.apimatic.io ).
 */
exports.__esModule = true;
exports.purchaseUnitSchema = void 0;
var schema_1 = require("../schema");
var amountWithBreakdown_1 = require("./amountWithBreakdown");
var item_1 = require("./item");
var payeeBase_1 = require("./payeeBase");
var paymentCollection_1 = require("./paymentCollection");
var paymentInstruction_1 = require("./paymentInstruction");
var shippingWithTrackingDetails_1 = require("./shippingWithTrackingDetails");
var supplementaryData_1 = require("./supplementaryData");
exports.purchaseUnitSchema = (0, schema_1.object)({
    referenceId: ['reference_id', (0, schema_1.optional)((0, schema_1.string)())],
    amount: ['amount', (0, schema_1.optional)((0, schema_1.lazy)(function () { return amountWithBreakdown_1.amountWithBreakdownSchema; }))],
    payee: ['payee', (0, schema_1.optional)((0, schema_1.lazy)(function () { return payeeBase_1.payeeBaseSchema; }))],
    paymentInstruction: [
        'payment_instruction',
        (0, schema_1.optional)((0, schema_1.lazy)(function () { return paymentInstruction_1.paymentInstructionSchema; })),
    ],
    description: ['description', (0, schema_1.optional)((0, schema_1.string)())],
    customId: ['custom_id', (0, schema_1.optional)((0, schema_1.string)())],
    invoiceId: ['invoice_id', (0, schema_1.optional)((0, schema_1.string)())],
    id: ['id', (0, schema_1.optional)((0, schema_1.string)())],
    softDescriptor: ['soft_descriptor', (0, schema_1.optional)((0, schema_1.string)())],
    items: ['items', (0, schema_1.optional)((0, schema_1.array)((0, schema_1.lazy)(function () { return item_1.itemSchema; })))],
    shipping: [
        'shipping',
        (0, schema_1.optional)((0, schema_1.lazy)(function () { return shippingWithTrackingDetails_1.shippingWithTrackingDetailsSchema; })),
    ],
    supplementaryData: [
        'supplementary_data',
        (0, schema_1.optional)((0, schema_1.lazy)(function () { return supplementaryData_1.supplementaryDataSchema; })),
    ],
    payments: ['payments', (0, schema_1.optional)((0, schema_1.lazy)(function () { return paymentCollection_1.paymentCollectionSchema; }))],
    mostRecentErrors: ['most_recent_errors', (0, schema_1.optional)((0, schema_1.array)((0, schema_1.unknown)()))]
});
