// @paypal/paypal-server-sdk/dist/cjs/models/sellerReceivableBreakdown.js
"use strict";
/**
 * PayPal Server SDKLib
 *
 * This file was automatically generated by APIMATIC v3.0 ( https://www.apimatic.io ).
 */
exports.__esModule = true;
exports.sellerReceivableBreakdownSchema = void 0;
var schema_1 = require("../schema");
var exchangeRate_1 = require("./exchangeRate");
var money_1 = require("./money");
var platformFee_1 = require("./platformFee");
exports.sellerReceivableBreakdownSchema = (0, schema_1.object)({
    grossAmount: ['gross_amount', (0, schema_1.lazy)(function () { return money_1.moneySchema; })],
    paypalFee: ['paypal_fee', (0, schema_1.optional)((0, schema_1.lazy)(function () { return money_1.moneySchema; }))],
    paypalFeeInReceivableCurrency: [
        'paypal_fee_in_receivable_currency',
        (0, schema_1.optional)((0, schema_1.lazy)(function () { return money_1.moneySchema; })),
    ],
    netAmount: ['net_amount', (0, schema_1.optional)((0, schema_1.lazy)(function () { return money_1.moneySchema; }))],
    receivableAmount: ['receivable_amount', (0, schema_1.optional)((0, schema_1.lazy)(function () { return money_1.moneySchema; }))],
    exchangeRate: ['exchange_rate', (0, schema_1.optional)((0, schema_1.lazy)(function () { return exchangeRate_1.exchangeRateSchema; }))],
    platformFees: [
        'platform_fees',
        (0, schema_1.optional)((0, schema_1.array)((0, schema_1.lazy)(function () { return platformFee_1.platformFeeSchema; }))),
    ]
});
