// @paypal/paypal-server-sdk/dist/cjs/models/setupTokenResponse.js
"use strict";
/**
 * PayPal Server SDKLib
 *
 * This file was automatically generated by APIMATIC v3.0 ( https://www.apimatic.io ).
 */
exports.__esModule = true;
exports.setupTokenResponseSchema = void 0;
var schema_1 = require("../schema");
var customer_1 = require("./customer");
var linkDescription_1 = require("./linkDescription");
var paymentTokenStatus_1 = require("./paymentTokenStatus");
var setupTokenResponsePaymentSource_1 = require("./setupTokenResponsePaymentSource");
exports.setupTokenResponseSchema = (0, schema_1.object)({
    id: ['id', (0, schema_1.optional)((0, schema_1.string)())],
    customer: ['customer', (0, schema_1.optional)((0, schema_1.lazy)(function () { return customer_1.customerSchema; }))],
    status: ['status', (0, schema_1.optional)(paymentTokenStatus_1.paymentTokenStatusSchema)],
    paymentSource: [
        'payment_source',
        (0, schema_1.optional)((0, schema_1.lazy)(function () { return setupTokenResponsePaymentSource_1.setupTokenResponsePaymentSourceSchema; })),
    ],
    links: ['links', (0, schema_1.optional)((0, schema_1.array)((0, schema_1.lazy)(function () { return linkDescription_1.linkDescriptionSchema; })))]
});
