// @paypal/paypal-server-sdk/dist/cjs/models/shipmentCarrier.js
"use strict";
/**
 * PayPal Server SDKLib
 *
 * This file was automatically generated by APIMATIC v3.0 ( https://www.apimatic.io ).
 */
exports.__esModule = true;
exports.shipmentCarrierSchema = exports.ShipmentCarrier = void 0;
var schema_1 = require("../schema");
/**
 * Enum for ShipmentCarrier
 */
var ShipmentCarrier;
(function (ShipmentCarrier) {
    ShipmentCarrier["DpdRu"] = "DPD_RU";
    ShipmentCarrier["BgBulgarianPost"] = "BG_BULGARIAN_POST";
    ShipmentCarrier["KrKoreaPost"] = "KR_KOREA_POST";
    ShipmentCarrier["ZaCourierit"] = "ZA_COURIERIT";
    ShipmentCarrier["FrExapaq"] = "FR_EXAPAQ";
    ShipmentCarrier["AreEmiratesPost"] = "ARE_EMIRATES_POST";
    ShipmentCarrier["Gac"] = "GAC";
    ShipmentCarrier["Geis"] = "GEIS";
    ShipmentCarrier["SfEx"] = "SF_EX";
    ShipmentCarrier["Pago"] = "PAGO";
    ShipmentCarrier["Myhermes"] = "MYHERMES";
    ShipmentCarrier["DiamondEurogistics"] = "DIAMOND_EUROGISTICS";
    ShipmentCarrier["CorporatecouriersWebhook"] = "CORPORATECOURIERS_WEBHOOK";
    ShipmentCarrier["Bond"] = "BOND";
    ShipmentCarrier["Omniparcel"] = "OMNIPARCEL";
    ShipmentCarrier["SkPosta"] = "SK_POSTA";
    ShipmentCarrier["Purolator"] = "PUROLATOR";
    ShipmentCarrier["FetchrWebhook"] = "FETCHR_WEBHOOK";
    ShipmentCarrier["Thedeliverygroup"] = "THEDELIVERYGROUP";
    ShipmentCarrier["CelloSquare"] = "CELLO_SQUARE";
    ShipmentCarrier["Tarrive"] = "TARRIVE";
    ShipmentCarrier["Collivery"] = "COLLIVERY";
    ShipmentCarrier["Mainfreight"] = "MAINFREIGHT";
    ShipmentCarrier["IndFirstflight"] = "IND_FIRSTFLIGHT";
    ShipmentCarrier["Acsworldwide"] = "ACSWORLDWIDE";
    ShipmentCarrier["Amstan"] = "AMSTAN";
    ShipmentCarrier["Okayparcel"] = "OKAYPARCEL";
    ShipmentCarrier["EnvialiaReference"] = "ENVIALIA_REFERENCE";
    ShipmentCarrier["SeurEs"] = "SEUR_ES";
    ShipmentCarrier["Continental"] = "CONTINENTAL";
    ShipmentCarrier["Fdsexpress"] = "FDSEXPRESS";
    ShipmentCarrier["AmazonFbaSwiship"] = "AMAZON_FBA_SWISHIP";
    ShipmentCarrier["Wyngs"] = "WYNGS";
    ShipmentCarrier["DhlActiveTracing"] = "DHL_ACTIVE_TRACING";
    ShipmentCarrier["Zyllem"] = "ZYLLEM";
    ShipmentCarrier["Ruston"] = "RUSTON";
    ShipmentCarrier["Xpost"] = "XPOST";
    ShipmentCarrier["CorreosEs"] = "CORREOS_ES";
    ShipmentCarrier["DhlFr"] = "DHL_FR";
    ShipmentCarrier["PanAsia"] = "PAN_ASIA";
    ShipmentCarrier["BrtIt"] = "BRT_IT";
    ShipmentCarrier["SreKorea"] = "SRE_KOREA";
    ShipmentCarrier["Speedee"] = "SPEEDEE";
    ShipmentCarrier["TntUk"] = "TNT_UK";
    ShipmentCarrier["Venipak"] = "VENIPAK";
    ShipmentCarrier["Shreenandancourier"] = "SHREENANDANCOURIER";
    ShipmentCarrier["Croshot"] = "CROSHOT";
    ShipmentCarrier["NipostNg"] = "NIPOST_NG";
    ShipmentCarrier["EpstGlbl"] = "EPST_GLBL";
    ShipmentCarrier["Newgistics"] = "NEWGISTICS";
    ShipmentCarrier["PostSlovenia"] = "POST_SLOVENIA";
    ShipmentCarrier["JerseyPost"] = "JERSEY_POST";
    ShipmentCarrier["Bombinoexp"] = "BOMBINOEXP";
    ShipmentCarrier["Wmg"] = "WMG";
    ShipmentCarrier["XqExpress"] = "XQ_EXPRESS";
    ShipmentCarrier["Furdeco"] = "FURDECO";
    ShipmentCarrier["LhtExpress"] = "LHT_EXPRESS";
    ShipmentCarrier["SouthAfricanPostOffice"] = "SOUTH_AFRICAN_POST_OFFICE";
    ShipmentCarrier["Spoton"] = "SPOTON";
    ShipmentCarrier["Dimerco"] = "DIMERCO";
    ShipmentCarrier["CyprusPostCyp"] = "CYPRUS_POST_CYP";
    ShipmentCarrier["Abcustom"] = "ABCUSTOM";
    ShipmentCarrier["IndDelivree"] = "IND_DELIVREE";
    ShipmentCarrier["CnBestexpress"] = "CN_BESTEXPRESS";
    ShipmentCarrier["DxSftp"] = "DX_SFTP";
    ShipmentCarrier["PickuppMys"] = "PICKUPP_MYS";
    ShipmentCarrier["Fmx"] = "FMX";
    ShipmentCarrier["Hellmann"] = "HELLMANN";
    ShipmentCarrier["ShipItAsia"] = "SHIP_IT_ASIA";
    ShipmentCarrier["KerryEcommerce"] = "KERRY_ECOMMERCE";
    ShipmentCarrier["Freterapido"] = "FRETERAPIDO";
    ShipmentCarrier["PitneyBowes"] = "PITNEY_BOWES";
    ShipmentCarrier["XpressenDk"] = "XPRESSEN_DK";
    ShipmentCarrier["SeurSpApi"] = "SEUR_SP_API";
    ShipmentCarrier["Deliveryontime"] = "DELIVERYONTIME";
    ShipmentCarrier["Jinsung"] = "JINSUNG";
    ShipmentCarrier["TransKargo"] = "TRANS_KARGO";
    ShipmentCarrier["SwishipDe"] = "SWISHIP_DE";
    ShipmentCarrier["IvoyWebhook"] = "IVOY_WEBHOOK";
    ShipmentCarrier["AirmeeWebhook"] = "AIRMEE_WEBHOOK";
    ShipmentCarrier["DhlBenelux"] = "DHL_BENELUX";
    ShipmentCarrier["Firstmile"] = "FIRSTMILE";
    ShipmentCarrier["FastwayIr"] = "FASTWAY_IR";
    ShipmentCarrier["HhExp"] = "HH_EXP";
    ShipmentCarrier["MysMypostOnline"] = "MYS_MYPOST_ONLINE";
    ShipmentCarrier["TntNl"] = "TNT_NL";
    ShipmentCarrier["Tipsa"] = "TIPSA";
    ShipmentCarrier["TaqbinMy"] = "TAQBIN_MY";
    ShipmentCarrier["Kgmhub"] = "KGMHUB";
    ShipmentCarrier["Intexpress"] = "INTEXPRESS";
    ShipmentCarrier["OverseExp"] = "OVERSE_EXP";
    ShipmentCarrier["Oneclick"] = "ONECLICK";
    ShipmentCarrier["RoadrunnerFreight"] = "ROADRUNNER_FREIGHT";
    ShipmentCarrier["GlsCrotia"] = "GLS_CROTIA";
    ShipmentCarrier["MrwFtp"] = "MRW_FTP";
    ShipmentCarrier["Bluex"] = "BLUEX";
    ShipmentCarrier["Dylt"] = "DYLT";
    ShipmentCarrier["DpdIr"] = "DPD_IR";
    ShipmentCarrier["SinGlbl"] = "SIN_GLBL";
    ShipmentCarrier["TuffnellsReference"] = "TUFFNELLS_REFERENCE";
    ShipmentCarrier["Cjpacket"] = "CJPACKET";
    ShipmentCarrier["Milkman"] = "MILKMAN";
    ShipmentCarrier["Asigna"] = "ASIGNA";
    ShipmentCarrier["Oneworldexpress"] = "ONEWORLDEXPRESS";
    ShipmentCarrier["RoyalMail"] = "ROYAL_MAIL";
    ShipmentCarrier["ViaExpress"] = "VIA_EXPRESS";
    ShipmentCarrier["Tigfreight"] = "TIGFREIGHT";
    ShipmentCarrier["ZtoExpress"] = "ZTO_EXPRESS";
    ShipmentCarrier["TwoGo"] = "TWO_GO";
    ShipmentCarrier["Iml"] = "IML";
    ShipmentCarrier["IntelValley"] = "INTEL_VALLEY";
    ShipmentCarrier["Efs"] = "EFS";
    ShipmentCarrier["UkUkMail"] = "UK_UK_MAIL";
    ShipmentCarrier["Ram"] = "RAM";
    ShipmentCarrier["Alliedexpress"] = "ALLIEDEXPRESS";
    ShipmentCarrier["ApcOvernight"] = "APC_OVERNIGHT";
    ShipmentCarrier["Shippit"] = "SHIPPIT";
    ShipmentCarrier["Tfm"] = "TFM";
    ShipmentCarrier["MXpress"] = "M_XPRESS";
    ShipmentCarrier["HdbBox"] = "HDB_BOX";
    ShipmentCarrier["ClevyLinks"] = "CLEVY_LINKS";
    ShipmentCarrier["Ibeone"] = "IBEONE";
    ShipmentCarrier["FiegeNl"] = "FIEGE_NL";
    ShipmentCarrier["KweGlobal"] = "KWE_GLOBAL";
    ShipmentCarrier["CtcExpress"] = "CTC_EXPRESS";
    ShipmentCarrier["Amazon"] = "AMAZON";
    ShipmentCarrier["MoreLink"] = "MORE_LINK";
    ShipmentCarrier["Jx"] = "JX";
    ShipmentCarrier["EasyMail"] = "EASY_MAIL";
    ShipmentCarrier["Aduiepyle"] = "ADUIEPYLE";
    ShipmentCarrier["GbPanther"] = "GB_PANTHER";
    ShipmentCarrier["Expresssale"] = "EXPRESSSALE";
    ShipmentCarrier["SgDetrack"] = "SG_DETRACK";
    ShipmentCarrier["TrunkrsWebhook"] = "TRUNKRS_WEBHOOK";
    ShipmentCarrier["Matdespatch"] = "MATDESPATCH";
    ShipmentCarrier["Dicom"] = "DICOM";
    ShipmentCarrier["Mbw"] = "MBW";
    ShipmentCarrier["KhmCambodiaPost"] = "KHM_CAMBODIA_POST";
    ShipmentCarrier["Sinotrans"] = "SINOTRANS";
    ShipmentCarrier["BrtItParcelid"] = "BRT_IT_PARCELID";
    ShipmentCarrier["DhlSupplyChain"] = "DHL_SUPPLY_CHAIN";
    ShipmentCarrier["DhlPl"] = "DHL_PL";
    ShipmentCarrier["Topyou"] = "TOPYOU";
    ShipmentCarrier["Palexpress"] = "PALEXPRESS";
    ShipmentCarrier["DhlSg"] = "DHL_SG";
    ShipmentCarrier["CnWedo"] = "CN_WEDO";
    ShipmentCarrier["Fulfillme"] = "FULFILLME";
    ShipmentCarrier["DpdDelistrack"] = "DPD_DELISTRACK";
    ShipmentCarrier["UpsReference"] = "UPS_REFERENCE";
    ShipmentCarrier["Caribou"] = "CARIBOU";
    ShipmentCarrier["LocusWebhook"] = "LOCUS_WEBHOOK";
    ShipmentCarrier["Dsv"] = "DSV";
    ShipmentCarrier["P2PTrc"] = "P2P_TRC";
    ShipmentCarrier["Directparcels"] = "DIRECTPARCELS";
    ShipmentCarrier["NovaPoshtaInt"] = "NOVA_POSHTA_INT";
    ShipmentCarrier["FedexPoland"] = "FEDEX_POLAND";
    ShipmentCarrier["CnJcex"] = "CN_JCEX";
    ShipmentCarrier["FarInternational"] = "FAR_INTERNATIONAL";
    ShipmentCarrier["Idexpress"] = "IDEXPRESS";
    ShipmentCarrier["Gangbao"] = "GANGBAO";
    ShipmentCarrier["Neway"] = "NEWAY";
    ShipmentCarrier["PostnlInt3S"] = "POSTNL_INT_3_S";
    ShipmentCarrier["RpxId"] = "RPX_ID";
    ShipmentCarrier["DesignertransportWebhook"] = "DESIGNERTRANSPORT_WEBHOOK";
    ShipmentCarrier["GlsSloven"] = "GLS_SLOVEN";
    ShipmentCarrier["ParcelledIn"] = "PARCELLED_IN";
    ShipmentCarrier["GsiExpress"] = "GSI_EXPRESS";
    ShipmentCarrier["ConWay"] = "CON_WAY";
    ShipmentCarrier["BrouwerTransport"] = "BROUWER_TRANSPORT";
    ShipmentCarrier["Cpex"] = "CPEX";
    ShipmentCarrier["IsraelPost"] = "ISRAEL_POST";
    ShipmentCarrier["DtdcIn"] = "DTDC_IN";
    ShipmentCarrier["PttPost"] = "PTT_POST";
    ShipmentCarrier["XdeWebhook"] = "XDE_WEBHOOK";
    ShipmentCarrier["Tolos"] = "TOLOS";
    ShipmentCarrier["GiaoHang"] = "GIAO_HANG";
    ShipmentCarrier["GeodisEspace"] = "GEODIS_ESPACE";
    ShipmentCarrier["MagyarHu"] = "MAGYAR_HU";
    ShipmentCarrier["DoordashWebhook"] = "DOORDASH_WEBHOOK";
    ShipmentCarrier["TikiId"] = "TIKI_ID";
    ShipmentCarrier["CjHkInternational"] = "CJ_HK_INTERNATIONAL";
    ShipmentCarrier["StarTrackExpress"] = "STAR_TRACK_EXPRESS";
    ShipmentCarrier["Helthjem"] = "HELTHJEM";
    ShipmentCarrier["Sfb2C"] = "SFB2C";
    ShipmentCarrier["Freightquote"] = "FREIGHTQUOTE";
    ShipmentCarrier["LandmarkGlobalReference"] = "LANDMARK_GLOBAL_REFERENCE";
    ShipmentCarrier["Parcel2Go"] = "PARCEL2GO";
    ShipmentCarrier["Delnext"] = "DELNEXT";
    ShipmentCarrier["Rcl"] = "RCL";
    ShipmentCarrier["CgsExpress"] = "CGS_EXPRESS";
    ShipmentCarrier["HkPost"] = "HK_POST";
    ShipmentCarrier["SapExpress"] = "SAP_EXPRESS";
    ShipmentCarrier["ParcelpostSg"] = "PARCELPOST_SG";
    ShipmentCarrier["Hermes"] = "HERMES";
    ShipmentCarrier["IndSafeexpress"] = "IND_SAFEEXPRESS";
    ShipmentCarrier["Tophatterexpress"] = "TOPHATTEREXPRESS";
    ShipmentCarrier["Mglobal"] = "MGLOBAL";
    ShipmentCarrier["Averitt"] = "AVERITT";
    ShipmentCarrier["Leader"] = "LEADER";
    ShipmentCarrier["_2Ebox"] = "_2EBOX";
    ShipmentCarrier["SgSpeedpost"] = "SG_SPEEDPOST";
    ShipmentCarrier["DbschenkerSe"] = "DBSCHENKER_SE";
    ShipmentCarrier["IsrPostDomestic"] = "ISR_POST_DOMESTIC";
    ShipmentCarrier["Bestwayparcel"] = "BESTWAYPARCEL";
    ShipmentCarrier["AsendiaDe"] = "ASENDIA_DE";
    ShipmentCarrier["NightlineUk"] = "NIGHTLINE_UK";
    ShipmentCarrier["TaqbinSg"] = "TAQBIN_SG";
    ShipmentCarrier["TckExpress"] = "TCK_EXPRESS";
    ShipmentCarrier["EndeavourDelivery"] = "ENDEAVOUR_DELIVERY";
    ShipmentCarrier["Nanjingwoyuan"] = "NANJINGWOYUAN";
    ShipmentCarrier["HeppnerFr"] = "HEPPNER_FR";
    ShipmentCarrier["EmpsCn"] = "EMPS_CN";
    ShipmentCarrier["Fonsen"] = "FONSEN";
    ShipmentCarrier["Pickrr"] = "PICKRR";
    ShipmentCarrier["ApcOvernightConnum"] = "APC_OVERNIGHT_CONNUM";
    ShipmentCarrier["StarTrackNextFlight"] = "STAR_TRACK_NEXT_FLIGHT";
    ShipmentCarrier["Dajin"] = "DAJIN";
    ShipmentCarrier["UpsFreight"] = "UPS_FREIGHT";
    ShipmentCarrier["PostaPlus"] = "POSTA_PLUS";
    ShipmentCarrier["Ceva"] = "CEVA";
    ShipmentCarrier["Anserx"] = "ANSERX";
    ShipmentCarrier["JsExpress"] = "JS_EXPRESS";
    ShipmentCarrier["Padtf"] = "PADTF";
    ShipmentCarrier["UpsMailInnovations"] = "UPS_MAIL_INNOVATIONS";
    ShipmentCarrier["Sypost"] = "SYPOST";
    ShipmentCarrier["AmazonShipMcf"] = "AMAZON_SHIP_MCF";
    ShipmentCarrier["Yusen"] = "YUSEN";
    ShipmentCarrier["Bring"] = "BRING";
    ShipmentCarrier["SdaIt"] = "SDA_IT";
    ShipmentCarrier["Gba"] = "GBA";
    ShipmentCarrier["Neweggexpress"] = "NEWEGGEXPRESS";
    ShipmentCarrier["SpeedcouriersGr"] = "SPEEDCOURIERS_GR";
    ShipmentCarrier["Forrun"] = "FORRUN";
    ShipmentCarrier["Pickup"] = "PICKUP";
    ShipmentCarrier["Ecms"] = "ECMS";
    ShipmentCarrier["Intelipost"] = "INTELIPOST";
    ShipmentCarrier["Flashexpress"] = "FLASHEXPRESS";
    ShipmentCarrier["CnSto"] = "CN_STO";
    ShipmentCarrier["SekoSftp"] = "SEKO_SFTP";
    ShipmentCarrier["HomeDeliverySolutions"] = "HOME_DELIVERY_SOLUTIONS";
    ShipmentCarrier["DpdHgry"] = "DPD_HGRY";
    ShipmentCarrier["KerryttcVn"] = "KERRYTTC_VN";
    ShipmentCarrier["JoyingBox"] = "JOYING_BOX";
    ShipmentCarrier["TotalExpress"] = "TOTAL_EXPRESS";
    ShipmentCarrier["ZjsExpress"] = "ZJS_EXPRESS";
    ShipmentCarrier["Starken"] = "STARKEN";
    ShipmentCarrier["Demandship"] = "DEMANDSHIP";
    ShipmentCarrier["CnDpex"] = "CN_DPEX";
    ShipmentCarrier["AupostCn"] = "AUPOST_CN";
    ShipmentCarrier["Logisters"] = "LOGISTERS";
    ShipmentCarrier["Goglobalpost"] = "GOGLOBALPOST";
    ShipmentCarrier["GlsCz"] = "GLS_CZ";
    ShipmentCarrier["PaackWebhook"] = "PAACK_WEBHOOK";
    ShipmentCarrier["GrabWebhook"] = "GRAB_WEBHOOK";
    ShipmentCarrier["Parcelpoint"] = "PARCELPOINT";
    ShipmentCarrier["Icumulus"] = "ICUMULUS";
    ShipmentCarrier["Daiglobaltrack"] = "DAIGLOBALTRACK";
    ShipmentCarrier["GlobalIparcel"] = "GLOBAL_IPARCEL";
    ShipmentCarrier["YurticiKargo"] = "YURTICI_KARGO";
    ShipmentCarrier["CnPaypalPackage"] = "CN_PAYPAL_PACKAGE";
    ShipmentCarrier["Parcel2Post"] = "PARCEL_2_POST";
    ShipmentCarrier["GlsIt"] = "GLS_IT";
    ShipmentCarrier["PilLogistics"] = "PIL_LOGISTICS";
    ShipmentCarrier["Heppner"] = "HEPPNER";
    ShipmentCarrier["GeneralOvernight"] = "GENERAL_OVERNIGHT";
    ShipmentCarrier["Happy2Point"] = "HAPPY2POINT";
    ShipmentCarrier["Chitchats"] = "CHITCHATS";
    ShipmentCarrier["Smooth"] = "SMOOTH";
    ShipmentCarrier["CleLogistics"] = "CLE_LOGISTICS";
    ShipmentCarrier["Fiege"] = "FIEGE";
    ShipmentCarrier["MxCargo"] = "MX_CARGO";
    ShipmentCarrier["Ziingfinalmile"] = "ZIINGFINALMILE";
    ShipmentCarrier["DaytonFreight"] = "DAYTON_FREIGHT";
    ShipmentCarrier["Tcs"] = "TCS";
    ShipmentCarrier["Aex"] = "AEX";
    ShipmentCarrier["HermesDe"] = "HERMES_DE";
    ShipmentCarrier["RoutificWebhook"] = "ROUTIFIC_WEBHOOK";
    ShipmentCarrier["Globavend"] = "GLOBAVEND";
    ShipmentCarrier["CjLogistics"] = "CJ_LOGISTICS";
    ShipmentCarrier["PalletNetwork"] = "PALLET_NETWORK";
    ShipmentCarrier["RafPh"] = "RAF_PH";
    ShipmentCarrier["UkXdp"] = "UK_XDP";
    ShipmentCarrier["PaperExpress"] = "PAPER_EXPRESS";
    ShipmentCarrier["LaPosteSuivi"] = "LA_POSTE_SUIVI";
    ShipmentCarrier["Paquetexpress"] = "PAQUETEXPRESS";
    ShipmentCarrier["Liefery"] = "LIEFERY";
    ShipmentCarrier["StreckTransport"] = "STRECK_TRANSPORT";
    ShipmentCarrier["PonyExpress"] = "PONY_EXPRESS";
    ShipmentCarrier["AlwaysExpress"] = "ALWAYS_EXPRESS";
    ShipmentCarrier["GbsBroker"] = "GBS_BROKER";
    ShipmentCarrier["CitylinkMy"] = "CITYLINK_MY";
    ShipmentCarrier["Alljoy"] = "ALLJOY";
    ShipmentCarrier["Yodel"] = "YODEL";
    ShipmentCarrier["YodelDir"] = "YODEL_DIR";
    ShipmentCarrier["Stone3Pl"] = "STONE3PL";
    ShipmentCarrier["ParcelpalWebhook"] = "PARCELPAL_WEBHOOK";
    ShipmentCarrier["DhlEcomerceAsa"] = "DHL_ECOMERCE_ASA";
    ShipmentCarrier["Simplypost"] = "SIMPLYPOST";
    ShipmentCarrier["KyExpress"] = "KY_EXPRESS";
    ShipmentCarrier["Shenzhen"] = "SHENZHEN";
    ShipmentCarrier["UsLasership"] = "US_LASERSHIP";
    ShipmentCarrier["UcExpre"] = "UC_EXPRE";
    ShipmentCarrier["Didadi"] = "DIDADI";
    ShipmentCarrier["CjKr"] = "CJ_KR";
    ShipmentCarrier["DbschenkerB2B"] = "DBSCHENKER_B2B";
    ShipmentCarrier["Mxe"] = "MXE";
    ShipmentCarrier["CaeDelivers"] = "CAE_DELIVERS";
    ShipmentCarrier["Pfcexpress"] = "PFCEXPRESS";
    ShipmentCarrier["Whistl"] = "WHISTL";
    ShipmentCarrier["Wepost"] = "WEPOST";
    ShipmentCarrier["DhlParcelEs"] = "DHL_PARCEL_ES";
    ShipmentCarrier["Ddexpress"] = "DDEXPRESS";
    ShipmentCarrier["AramexAu"] = "ARAMEX_AU";
    ShipmentCarrier["Bneed"] = "BNEED";
    ShipmentCarrier["HkTgx"] = "HK_TGX";
    ShipmentCarrier["LatvijasPasts"] = "LATVIJAS_PASTS";
    ShipmentCarrier["Viaeurope"] = "VIAEUROPE";
    ShipmentCarrier["CorreoUy"] = "CORREO_UY";
    ShipmentCarrier["ChronopostFr"] = "CHRONOPOST_FR";
    ShipmentCarrier["JNet"] = "J_NET";
    ShipmentCarrier["_6Ls"] = "_6LS";
    ShipmentCarrier["BlrBelpost"] = "BLR_BELPOST";
    ShipmentCarrier["Birdsystem"] = "BIRDSYSTEM";
    ShipmentCarrier["Dobropost"] = "DOBROPOST";
    ShipmentCarrier["WahanaId"] = "WAHANA_ID";
    ShipmentCarrier["Weaship"] = "WEASHIP";
    ShipmentCarrier["Sonictl"] = "SONICTL";
    ShipmentCarrier["Kwt"] = "KWT";
    ShipmentCarrier["AfllogFtp"] = "AFLLOG_FTP";
    ShipmentCarrier["SkynetWorldwide"] = "SKYNET_WORLDWIDE";
    ShipmentCarrier["NovaPoshta"] = "NOVA_POSHTA";
    ShipmentCarrier["Seino"] = "SEINO";
    ShipmentCarrier["Szendex"] = "SZENDEX";
    ShipmentCarrier["BpostInt"] = "BPOST_INT";
    ShipmentCarrier["DbschenkerSv"] = "DBSCHENKER_SV";
    ShipmentCarrier["AoDeutschland"] = "AO_DEUTSCHLAND";
    ShipmentCarrier["EuFleetSolutions"] = "EU_FLEET_SOLUTIONS";
    ShipmentCarrier["Pcfcorp"] = "PCFCORP";
    ShipmentCarrier["Linkbridge"] = "LINKBRIDGE";
    ShipmentCarrier["Primamulticipta"] = "PRIMAMULTICIPTA";
    ShipmentCarrier["Courex"] = "COUREX";
    ShipmentCarrier["ZajilExpress"] = "ZAJIL_EXPRESS";
    ShipmentCarrier["Collectco"] = "COLLECTCO";
    ShipmentCarrier["Jtexpress"] = "JTEXPRESS";
    ShipmentCarrier["FedexUk"] = "FEDEX_UK";
    ShipmentCarrier["Uship"] = "USHIP";
    ShipmentCarrier["Pixsell"] = "PIXSELL";
    ShipmentCarrier["Shiptor"] = "SHIPTOR";
    ShipmentCarrier["Cdek"] = "CDEK";
    ShipmentCarrier["VnmViettelpost"] = "VNM_VIETTELPOST";
    ShipmentCarrier["CjCentury"] = "CJ_CENTURY";
    ShipmentCarrier["Gso"] = "GSO";
    ShipmentCarrier["Viwo"] = "VIWO";
    ShipmentCarrier["Skybox"] = "SKYBOX";
    ShipmentCarrier["Kerrytj"] = "KERRYTJ";
    ShipmentCarrier["NtlogisticsVn"] = "NTLOGISTICS_VN";
    ShipmentCarrier["SdhScm"] = "SDH_SCM";
    ShipmentCarrier["Zinc"] = "ZINC";
    ShipmentCarrier["DpeSouthAfrc"] = "DPE_SOUTH_AFRC";
    ShipmentCarrier["CeskaCz"] = "CESKA_CZ";
    ShipmentCarrier["AcsGr"] = "ACS_GR";
    ShipmentCarrier["Dealersend"] = "DEALERSEND";
    ShipmentCarrier["Jocom"] = "JOCOM";
    ShipmentCarrier["Cse"] = "CSE";
    ShipmentCarrier["TforceFinalmile"] = "TFORCE_FINALMILE";
    ShipmentCarrier["ShipGate"] = "SHIP_GATE";
    ShipmentCarrier["Shipter"] = "SHIPTER";
    ShipmentCarrier["NationalSameday"] = "NATIONAL_SAMEDAY";
    ShipmentCarrier["Yunexpress"] = "YUNEXPRESS";
    ShipmentCarrier["Cainiao"] = "CAINIAO";
    ShipmentCarrier["DmsMatrix"] = "DMS_MATRIX";
    ShipmentCarrier["Directlog"] = "DIRECTLOG";
    ShipmentCarrier["AsendiaUs"] = "ASENDIA_US";
    ShipmentCarrier["_3Jmslogistics"] = "_3JMSLOGISTICS";
    ShipmentCarrier["LiccardiExpress"] = "LICCARDI_EXPRESS";
    ShipmentCarrier["SkyPostal"] = "SKY_POSTAL";
    ShipmentCarrier["Cnwangtong"] = "CNWANGTONG";
    ShipmentCarrier["PostnordLogisticsDk"] = "POSTNORD_LOGISTICS_DK";
    ShipmentCarrier["Logistika"] = "LOGISTIKA";
    ShipmentCarrier["Celeritas"] = "CELERITAS";
    ShipmentCarrier["Pressiode"] = "PRESSIODE";
    ShipmentCarrier["ShreeMaruti"] = "SHREE_MARUTI";
    ShipmentCarrier["LogisticsworldwideHk"] = "LOGISTICSWORLDWIDE_HK";
    ShipmentCarrier["Efex"] = "EFEX";
    ShipmentCarrier["Lotte"] = "LOTTE";
    ShipmentCarrier["Lonestar"] = "LONESTAR";
    ShipmentCarrier["Aprisaexpress"] = "APRISAEXPRESS";
    ShipmentCarrier["BelRs"] = "BEL_RS";
    ShipmentCarrier["OsmWorldwide"] = "OSM_WORLDWIDE";
    ShipmentCarrier["WestgateGl"] = "WESTGATE_GL";
    ShipmentCarrier["Fastrack"] = "FASTRACK";
    ShipmentCarrier["DtdExpr"] = "DTD_EXPR";
    ShipmentCarrier["Alfatrex"] = "ALFATREX";
    ShipmentCarrier["Promeddelivery"] = "PROMEDDELIVERY";
    ShipmentCarrier["ThabitLogistics"] = "THABIT_LOGISTICS";
    ShipmentCarrier["HctLogistics"] = "HCT_LOGISTICS";
    ShipmentCarrier["CarryFlap"] = "CARRY_FLAP";
    ShipmentCarrier["UsOldDominion"] = "US_OLD_DOMINION";
    ShipmentCarrier["AnicamBox"] = "ANICAM_BOX";
    ShipmentCarrier["Wanbexpress"] = "WANBEXPRESS";
    ShipmentCarrier["AnPost"] = "AN_POST";
    ShipmentCarrier["DpdLocal"] = "DPD_LOCAL";
    ShipmentCarrier["Stallionexpress"] = "STALLIONEXPRESS";
    ShipmentCarrier["Raiderex"] = "RAIDEREX";
    ShipmentCarrier["Shopfans"] = "SHOPFANS";
    ShipmentCarrier["KyungdongParcel"] = "KYUNGDONG_PARCEL";
    ShipmentCarrier["ChampionLogistics"] = "CHAMPION_LOGISTICS";
    ShipmentCarrier["PickuppSgp"] = "PICKUPP_SGP";
    ShipmentCarrier["MorningExpress"] = "MORNING_EXPRESS";
    ShipmentCarrier["Nacex"] = "NACEX";
    ShipmentCarrier["ThenileWebhook"] = "THENILE_WEBHOOK";
    ShipmentCarrier["Holisol"] = "HOLISOL";
    ShipmentCarrier["LbcexpressFtp"] = "LBCEXPRESS_FTP";
    ShipmentCarrier["Kurasi"] = "KURASI";
    ShipmentCarrier["UsfReddaway"] = "USF_REDDAWAY";
    ShipmentCarrier["Apg"] = "APG";
    ShipmentCarrier["CnBoxc"] = "CN_BOXC";
    ShipmentCarrier["Ecoscooting"] = "ECOSCOOTING";
    ShipmentCarrier["Mainway"] = "MAINWAY";
    ShipmentCarrier["Paperfly"] = "PAPERFLY";
    ShipmentCarrier["Houndexpress"] = "HOUNDEXPRESS";
    ShipmentCarrier["BoxBerry"] = "BOX_BERRY";
    ShipmentCarrier["EpBox"] = "EP_BOX";
    ShipmentCarrier["PlusLogUk"] = "PLUS_LOG_UK";
    ShipmentCarrier["Fulfilla"] = "FULFILLA";
    ShipmentCarrier["Ase"] = "ASE";
    ShipmentCarrier["MailPlus"] = "MAIL_PLUS";
    ShipmentCarrier["XpoLogistics"] = "XPO_LOGISTICS";
    ShipmentCarrier["Wndirect"] = "WNDIRECT";
    ShipmentCarrier["CloudwishAsia"] = "CLOUDWISH_ASIA";
    ShipmentCarrier["Zeleris"] = "ZELERIS";
    ShipmentCarrier["GioExpress"] = "GIO_EXPRESS";
    ShipmentCarrier["OcsWorldwide"] = "OCS_WORLDWIDE";
    ShipmentCarrier["ArkLogistics"] = "ARK_LOGISTICS";
    ShipmentCarrier["Aquiline"] = "AQUILINE";
    ShipmentCarrier["PilotFreight"] = "PILOT_FREIGHT";
    ShipmentCarrier["Qwintry"] = "QWINTRY";
    ShipmentCarrier["DanskeFragt"] = "DANSKE_FRAGT";
    ShipmentCarrier["Carriers"] = "CARRIERS";
    ShipmentCarrier["AirCanadaGlobal"] = "AIR_CANADA_GLOBAL";
    ShipmentCarrier["PresidentTrans"] = "PRESIDENT_TRANS";
    ShipmentCarrier["Stepforwardfs"] = "STEPFORWARDFS";
    ShipmentCarrier["SkynetUk"] = "SKYNET_UK";
    ShipmentCarrier["Pittohio"] = "PITTOHIO";
    ShipmentCarrier["CorreosExpress"] = "CORREOS_EXPRESS";
    ShipmentCarrier["RlUs"] = "RL_US";
    ShipmentCarrier["Destiny"] = "DESTINY";
    ShipmentCarrier["UkYodel"] = "UK_YODEL";
    ShipmentCarrier["CometTech"] = "COMET_TECH";
    ShipmentCarrier["DhlParcelRu"] = "DHL_PARCEL_RU";
    ShipmentCarrier["TntRefr"] = "TNT_REFR";
    ShipmentCarrier["ShreeAnjaniCourier"] = "SHREE_ANJANI_COURIER";
    ShipmentCarrier["MikropakketBe"] = "MIKROPAKKET_BE";
    ShipmentCarrier["EtsExpress"] = "ETS_EXPRESS";
    ShipmentCarrier["ColisPrive"] = "COLIS_PRIVE";
    ShipmentCarrier["CnYunda"] = "CN_YUNDA";
    ShipmentCarrier["AaaCooper"] = "AAA_COOPER";
    ShipmentCarrier["RocketParcel"] = "ROCKET_PARCEL";
    ShipmentCarrier["_360Lion"] = "_360LION";
    ShipmentCarrier["Pandu"] = "PANDU";
    ShipmentCarrier["ProfessionalCouriers"] = "PROFESSIONAL_COURIERS";
    ShipmentCarrier["Flytexpress"] = "FLYTEXPRESS";
    ShipmentCarrier["LogisticsworldwideMy"] = "LOGISTICSWORLDWIDE_MY";
    ShipmentCarrier["CorreosDeEspana"] = "CORREOS_DE_ESPANA";
    ShipmentCarrier["Imx"] = "IMX";
    ShipmentCarrier["FourPxExpress"] = "FOUR_PX_EXPRESS";
    ShipmentCarrier["Xpressbees"] = "XPRESSBEES";
    ShipmentCarrier["PickuppVnm"] = "PICKUPP_VNM";
    ShipmentCarrier["StartrackExpress1"] = "STARTRACK_EXPRESS";
    ShipmentCarrier["FrColissimo"] = "FR_COLISSIMO";
    ShipmentCarrier["NacexSpainReference"] = "NACEX_SPAIN_REFERENCE";
    ShipmentCarrier["DhlSupplyChainAu"] = "DHL_SUPPLY_CHAIN_AU";
    ShipmentCarrier["Eshipping"] = "ESHIPPING";
    ShipmentCarrier["Shreetirupati"] = "SHREETIRUPATI";
    ShipmentCarrier["HxExpress"] = "HX_EXPRESS";
    ShipmentCarrier["Indopaket"] = "INDOPAKET";
    ShipmentCarrier["Cn17Post"] = "CN_17POST";
    ShipmentCarrier["K1Express"] = "K1_EXPRESS";
    ShipmentCarrier["CjGls"] = "CJ_GLS";
    ShipmentCarrier["MysGdex"] = "MYS_GDEX";
    ShipmentCarrier["Nationex"] = "NATIONEX";
    ShipmentCarrier["Anjun"] = "ANJUN";
    ShipmentCarrier["Fargood"] = "FARGOOD";
    ShipmentCarrier["SmgExpress"] = "SMG_EXPRESS";
    ShipmentCarrier["Rzyexpress"] = "RZYEXPRESS";
    ShipmentCarrier["Sefl"] = "SEFL";
    ShipmentCarrier["TntClickIt"] = "TNT_CLICK_IT";
    ShipmentCarrier["Hdb"] = "HDB";
    ShipmentCarrier["Hipshipper"] = "HIPSHIPPER";
    ShipmentCarrier["Rpxlogistics"] = "RPXLOGISTICS";
    ShipmentCarrier["Kuehne"] = "KUEHNE";
    ShipmentCarrier["ItNexive"] = "IT_NEXIVE";
    ShipmentCarrier["Pts"] = "PTS";
    ShipmentCarrier["SwissPostFtp"] = "SWISS_POST_FTP";
    ShipmentCarrier["FastrkServ"] = "FASTRK_SERV";
    ShipmentCarrier["_472"] = "_4_72";
    ShipmentCarrier["UsYrc"] = "US_YRC";
    ShipmentCarrier["PostnlIntl3S"] = "POSTNL_INTL_3S";
    ShipmentCarrier["ElianPost"] = "ELIAN_POST";
    ShipmentCarrier["Cubyn"] = "CUBYN";
    ShipmentCarrier["SauSaudiPost"] = "SAU_SAUDI_POST";
    ShipmentCarrier["AbxexpressMy"] = "ABXEXPRESS_MY";
    ShipmentCarrier["HuahanExpress"] = "HUAHAN_EXPRESS";
    ShipmentCarrier["ZesExpress"] = "ZES_EXPRESS";
    ShipmentCarrier["ZeptoExpress"] = "ZEPTO_EXPRESS";
    ShipmentCarrier["SkynetZa"] = "SKYNET_ZA";
    ShipmentCarrier["Zeek2Door"] = "ZEEK_2_DOOR";
    ShipmentCarrier["Blinklastmile"] = "BLINKLASTMILE";
    ShipmentCarrier["PostaUkr"] = "POSTA_UKR";
    ShipmentCarrier["Chrobinson"] = "CHROBINSON";
    ShipmentCarrier["CnPost56"] = "CN_POST56";
    ShipmentCarrier["CourantPlus"] = "COURANT_PLUS";
    ShipmentCarrier["ScudexExpress"] = "SCUDEX_EXPRESS";
    ShipmentCarrier["Shipentegra"] = "SHIPENTEGRA";
    ShipmentCarrier["BTwoCEurope"] = "B_TWO_C_EUROPE";
    ShipmentCarrier["Cope"] = "COPE";
    ShipmentCarrier["IndGati"] = "IND_GATI";
    ShipmentCarrier["CnWishpost"] = "CN_WISHPOST";
    ShipmentCarrier["NacexEs"] = "NACEX_ES";
    ShipmentCarrier["TaqbinHk"] = "TAQBIN_HK";
    ShipmentCarrier["Globaltranz"] = "GLOBALTRANZ";
    ShipmentCarrier["Hkd"] = "HKD";
    ShipmentCarrier["Bjshomedelivery"] = "BJSHOMEDELIVERY";
    ShipmentCarrier["Omniva"] = "OMNIVA";
    ShipmentCarrier["Sutton"] = "SUTTON";
    ShipmentCarrier["PantherReference"] = "PANTHER_REFERENCE";
    ShipmentCarrier["Sfcservice"] = "SFCSERVICE";
    ShipmentCarrier["Ltl"] = "LTL";
    ShipmentCarrier["Parknparcel"] = "PARKNPARCEL";
    ShipmentCarrier["SpringGds"] = "SPRING_GDS";
    ShipmentCarrier["Ecexpress"] = "ECEXPRESS";
    ShipmentCarrier["InterparcelAu"] = "INTERPARCEL_AU";
    ShipmentCarrier["Agility"] = "AGILITY";
    ShipmentCarrier["XlExpress"] = "XL_EXPRESS";
    ShipmentCarrier["Aderonline"] = "ADERONLINE";
    ShipmentCarrier["Directcouriers"] = "DIRECTCOURIERS";
    ShipmentCarrier["Planzer"] = "PLANZER";
    ShipmentCarrier["Sending"] = "SENDING";
    ShipmentCarrier["NinjavanWb"] = "NINJAVAN_WB";
    ShipmentCarrier["NationwideMy"] = "NATIONWIDE_MY";
    ShipmentCarrier["Sendit"] = "SENDIT";
    ShipmentCarrier["GbArrow"] = "GB_ARROW";
    ShipmentCarrier["IndGojavas"] = "IND_GOJAVAS";
    ShipmentCarrier["Kpost"] = "KPOST";
    ShipmentCarrier["DhlFreight"] = "DHL_FREIGHT";
    ShipmentCarrier["Bluecare"] = "BLUECARE";
    ShipmentCarrier["Jindouyun"] = "JINDOUYUN";
    ShipmentCarrier["Trackon"] = "TRACKON";
    ShipmentCarrier["GbTuffnells"] = "GB_TUFFNELLS";
    ShipmentCarrier["Trumpcard"] = "TRUMPCARD";
    ShipmentCarrier["Etotal"] = "ETOTAL";
    ShipmentCarrier["SfplusWebhook"] = "SFPLUS_WEBHOOK";
    ShipmentCarrier["Sekologistics"] = "SEKOLOGISTICS";
    ShipmentCarrier["Hermes2MannHandling"] = "HERMES_2MANN_HANDLING";
    ShipmentCarrier["DpdLocalRef"] = "DPD_LOCAL_REF";
    ShipmentCarrier["Uds"] = "UDS";
    ShipmentCarrier["ZaSpecialisedFreight"] = "ZA_SPECIALISED_FREIGHT";
    ShipmentCarrier["ThaKerry"] = "THA_KERRY";
    ShipmentCarrier["PrtIntSeur"] = "PRT_INT_SEUR";
    ShipmentCarrier["BraCorreios"] = "BRA_CORREIOS";
    ShipmentCarrier["NzNzPost"] = "NZ_NZ_POST";
    ShipmentCarrier["CnEquick"] = "CN_EQUICK";
    ShipmentCarrier["MysEms"] = "MYS_EMS";
    ShipmentCarrier["GbNorsk"] = "GB_NORSK";
    ShipmentCarrier["EspMrw"] = "ESP_MRW";
    ShipmentCarrier["EspPacklink"] = "ESP_PACKLINK";
    ShipmentCarrier["KangarooMy"] = "KANGAROO_MY";
    ShipmentCarrier["Rpx"] = "RPX";
    ShipmentCarrier["XdpUkReference"] = "XDP_UK_REFERENCE";
    ShipmentCarrier["NinjavanMy"] = "NINJAVAN_MY";
    ShipmentCarrier["Adicional"] = "ADICIONAL";
    ShipmentCarrier["Roadbull"] = "ROADBULL";
    ShipmentCarrier["Yakit"] = "YAKIT";
    ShipmentCarrier["Mailamericas"] = "MAILAMERICAS";
    ShipmentCarrier["Mikropakket"] = "MIKROPAKKET";
    ShipmentCarrier["Dynalogic"] = "DYNALOGIC";
    ShipmentCarrier["DhlEs"] = "DHL_ES";
    ShipmentCarrier["DhlParcelNl"] = "DHL_PARCEL_NL";
    ShipmentCarrier["DhlGlobalMailAsia"] = "DHL_GLOBAL_MAIL_ASIA";
    ShipmentCarrier["DawnWing"] = "DAWN_WING";
    ShipmentCarrier["GenikiGr"] = "GENIKI_GR";
    ShipmentCarrier["HermesworldUk"] = "HERMESWORLD_UK";
    ShipmentCarrier["Alphafast"] = "ALPHAFAST";
    ShipmentCarrier["Buylogic"] = "BUYLOGIC";
    ShipmentCarrier["Ekart"] = "EKART";
    ShipmentCarrier["MexSenda"] = "MEX_SENDA";
    ShipmentCarrier["SfcLogistics"] = "SFC_LOGISTICS";
    ShipmentCarrier["PostSerbia"] = "POST_SERBIA";
    ShipmentCarrier["IndDelhivery"] = "IND_DELHIVERY";
    ShipmentCarrier["DeDpdDelistrack"] = "DE_DPD_DELISTRACK";
    ShipmentCarrier["Rpd2Man"] = "RPD2MAN";
    ShipmentCarrier["CnSfExpress"] = "CN_SF_EXPRESS";
    ShipmentCarrier["Yanwen"] = "YANWEN";
    ShipmentCarrier["MysSkynet"] = "MYS_SKYNET";
    ShipmentCarrier["CorreosDeMexico"] = "CORREOS_DE_MEXICO";
    ShipmentCarrier["CblLogistica"] = "CBL_LOGISTICA";
    ShipmentCarrier["MexEstafeta"] = "MEX_ESTAFETA";
    ShipmentCarrier["AuAustrianPost"] = "AU_AUSTRIAN_POST";
    ShipmentCarrier["Rincos"] = "RINCOS";
    ShipmentCarrier["NldDhl"] = "NLD_DHL";
    ShipmentCarrier["RussianPost"] = "RUSSIAN_POST";
    ShipmentCarrier["CouriersPlease"] = "COURIERS_PLEASE";
    ShipmentCarrier["PostnordLogistics"] = "POSTNORD_LOGISTICS";
    ShipmentCarrier["Fedex"] = "FEDEX";
    ShipmentCarrier["DpeExpress"] = "DPE_EXPRESS";
    ShipmentCarrier["Dpd"] = "DPD";
    ShipmentCarrier["Adsone"] = "ADSONE";
    ShipmentCarrier["IdnJne"] = "IDN_JNE";
    ShipmentCarrier["Thecourierguy"] = "THECOURIERGUY";
    ShipmentCarrier["Cnexps"] = "CNEXPS";
    ShipmentCarrier["PrtChronopost"] = "PRT_CHRONOPOST";
    ShipmentCarrier["LandmarkGlobal"] = "LANDMARK_GLOBAL";
    ShipmentCarrier["ItDhlEcommerce"] = "IT_DHL_ECOMMERCE";
    ShipmentCarrier["EspNacex"] = "ESP_NACEX";
    ShipmentCarrier["PrtCtt"] = "PRT_CTT";
    ShipmentCarrier["BeKiala"] = "BE_KIALA";
    ShipmentCarrier["AsendiaUk"] = "ASENDIA_UK";
    ShipmentCarrier["GlobalTnt"] = "GLOBAL_TNT";
    ShipmentCarrier["PosturIs"] = "POSTUR_IS";
    ShipmentCarrier["EparcelKr"] = "EPARCEL_KR";
    ShipmentCarrier["InpostPaczkomaty"] = "INPOST_PACZKOMATY";
    ShipmentCarrier["ItPosteItalia"] = "IT_POSTE_ITALIA";
    ShipmentCarrier["BeBpost"] = "BE_BPOST";
    ShipmentCarrier["PlPocztaPolska"] = "PL_POCZTA_POLSKA";
    ShipmentCarrier["MysMysPost"] = "MYS_MYS_POST";
    ShipmentCarrier["SgSgPost"] = "SG_SG_POST";
    ShipmentCarrier["ThaThailandPost"] = "THA_THAILAND_POST";
    ShipmentCarrier["Lexship"] = "LEXSHIP";
    ShipmentCarrier["FastwayNz"] = "FASTWAY_NZ";
    ShipmentCarrier["DhlAu"] = "DHL_AU";
    ShipmentCarrier["Costmeticsnow"] = "COSTMETICSNOW";
    ShipmentCarrier["Pflogistics"] = "PFLOGISTICS";
    ShipmentCarrier["LoomisExpress"] = "LOOMIS_EXPRESS";
    ShipmentCarrier["GlsItaly"] = "GLS_ITALY";
    ShipmentCarrier["Line"] = "LINE";
    ShipmentCarrier["GelExpress"] = "GEL_EXPRESS";
    ShipmentCarrier["Huodull"] = "HUODULL";
    ShipmentCarrier["NinjavanSg"] = "NINJAVAN_SG";
    ShipmentCarrier["Janio"] = "JANIO";
    ShipmentCarrier["AoCourier"] = "AO_COURIER";
    ShipmentCarrier["BrtItSenderRef"] = "BRT_IT_SENDER_REF";
    ShipmentCarrier["Sailpost"] = "SAILPOST";
    ShipmentCarrier["Lalamove"] = "LALAMOVE";
    ShipmentCarrier["NewzealandCouriers"] = "NEWZEALAND_COURIERS";
    ShipmentCarrier["Etomars"] = "ETOMARS";
    ShipmentCarrier["Virtransport"] = "VIRTRANSPORT";
    ShipmentCarrier["Wizmo"] = "WIZMO";
    ShipmentCarrier["Palletways"] = "PALLETWAYS";
    ShipmentCarrier["IDika"] = "I_DIKA";
    ShipmentCarrier["CflLogistics"] = "CFL_LOGISTICS";
    ShipmentCarrier["Gemworldwide"] = "GEMWORLDWIDE";
    ShipmentCarrier["GlobalExpress"] = "GLOBAL_EXPRESS";
    ShipmentCarrier["LogistyxTransgroup"] = "LOGISTYX_TRANSGROUP";
    ShipmentCarrier["WestbankCourier"] = "WESTBANK_COURIER";
    ShipmentCarrier["ArcoSpedizioni"] = "ARCO_SPEDIZIONI";
    ShipmentCarrier["YdhExpress"] = "YDH_EXPRESS";
    ShipmentCarrier["Parcelinklogistics"] = "PARCELINKLOGISTICS";
    ShipmentCarrier["Cndexpress"] = "CNDEXPRESS";
    ShipmentCarrier["NoxNightTimeExpress"] = "NOX_NIGHT_TIME_EXPRESS";
    ShipmentCarrier["Aeronet"] = "AERONET";
    ShipmentCarrier["Ltianexp"] = "LTIANEXP";
    ShipmentCarrier["Integra2Ftp"] = "INTEGRA2_FTP";
    ShipmentCarrier["Parcelone"] = "PARCELONE";
    ShipmentCarrier["NoxNachtexpress"] = "NOX_NACHTEXPRESS";
    ShipmentCarrier["CnChinaPostEms"] = "CN_CHINA_POST_EMS";
    ShipmentCarrier["Chukou1"] = "CHUKOU1";
    ShipmentCarrier["GlsSlov"] = "GLS_SLOV";
    ShipmentCarrier["OrangeDs"] = "ORANGE_DS";
    ShipmentCarrier["JoomLogis"] = "JOOM_LOGIS";
    ShipmentCarrier["AusStartrack"] = "AUS_STARTRACK";
    ShipmentCarrier["Dhl"] = "DHL";
    ShipmentCarrier["GbApc"] = "GB_APC";
    ShipmentCarrier["Bondscouriers"] = "BONDSCOURIERS";
    ShipmentCarrier["JpnJapanPost"] = "JPN_JAPAN_POST";
    ShipmentCarrier["Usps"] = "USPS";
    ShipmentCarrier["Winit"] = "WINIT";
    ShipmentCarrier["ArgOca"] = "ARG_OCA";
    ShipmentCarrier["TwTaiwanPost"] = "TW_TAIWAN_POST";
    ShipmentCarrier["DmmNetwork"] = "DMM_NETWORK";
    ShipmentCarrier["Tnt"] = "TNT";
    ShipmentCarrier["BhPosta"] = "BH_POSTA";
    ShipmentCarrier["SwePostnord"] = "SWE_POSTNORD";
    ShipmentCarrier["CaCanadaPost"] = "CA_CANADA_POST";
    ShipmentCarrier["Wiseloads"] = "WISELOADS";
    ShipmentCarrier["AsendiaHk"] = "ASENDIA_HK";
    ShipmentCarrier["NldGls"] = "NLD_GLS";
    ShipmentCarrier["MexRedpack"] = "MEX_REDPACK";
    ShipmentCarrier["JetShip"] = "JET_SHIP";
    ShipmentCarrier["DeDhlExpress"] = "DE_DHL_EXPRESS";
    ShipmentCarrier["NinjavanThai"] = "NINJAVAN_THAI";
    ShipmentCarrier["RabenGroup"] = "RABEN_GROUP";
    ShipmentCarrier["EspAsm"] = "ESP_ASM";
    ShipmentCarrier["HrvHrvatska"] = "HRV_HRVATSKA";
    ShipmentCarrier["GlobalEstes"] = "GLOBAL_ESTES";
    ShipmentCarrier["LtuLietuvos"] = "LTU_LIETUVOS";
    ShipmentCarrier["BelDhl"] = "BEL_DHL";
    ShipmentCarrier["AuAuPost"] = "AU_AU_POST";
    ShipmentCarrier["Speedexcourier"] = "SPEEDEXCOURIER";
    ShipmentCarrier["FrColis"] = "FR_COLIS";
    ShipmentCarrier["Aramex"] = "ARAMEX";
    ShipmentCarrier["Dpex"] = "DPEX";
    ShipmentCarrier["MysAirpak"] = "MYS_AIRPAK";
    ShipmentCarrier["Cuckooexpress"] = "CUCKOOEXPRESS";
    ShipmentCarrier["DpdPoland"] = "DPD_POLAND";
    ShipmentCarrier["NldPostnl"] = "NLD_POSTNL";
    ShipmentCarrier["NimExpress"] = "NIM_EXPRESS";
    ShipmentCarrier["Quantium"] = "QUANTIUM";
    ShipmentCarrier["Sendle"] = "SENDLE";
    ShipmentCarrier["EspRedur"] = "ESP_REDUR";
    ShipmentCarrier["Matkahuolto"] = "MATKAHUOLTO";
    ShipmentCarrier["Cpacket"] = "CPACKET";
    ShipmentCarrier["Posti"] = "POSTI";
    ShipmentCarrier["HunterExpress"] = "HUNTER_EXPRESS";
    ShipmentCarrier["ChoirExp"] = "CHOIR_EXP";
    ShipmentCarrier["LegionExpress"] = "LEGION_EXPRESS";
    ShipmentCarrier["AustrianPostExpress"] = "AUSTRIAN_POST_EXPRESS";
    ShipmentCarrier["Grupo"] = "GRUPO";
    ShipmentCarrier["PostaRo"] = "POSTA_RO";
    ShipmentCarrier["InterparcelUk"] = "INTERPARCEL_UK";
    ShipmentCarrier["GlobalAbf"] = "GLOBAL_ABF";
    ShipmentCarrier["PostenNorge"] = "POSTEN_NORGE";
    ShipmentCarrier["XpertDelivery"] = "XPERT_DELIVERY";
    ShipmentCarrier["DhlRefr"] = "DHL_REFR";
    ShipmentCarrier["DhlHk"] = "DHL_HK";
    ShipmentCarrier["SkynetUae"] = "SKYNET_UAE";
    ShipmentCarrier["Gojek"] = "GOJEK";
    ShipmentCarrier["YodelIntnl"] = "YODEL_INTNL";
    ShipmentCarrier["Janco"] = "JANCO";
    ShipmentCarrier["Yto"] = "YTO";
    ShipmentCarrier["WiseExpress"] = "WISE_EXPRESS";
    ShipmentCarrier["JtexpressVn"] = "JTEXPRESS_VN";
    ShipmentCarrier["FedexIntlMlserv"] = "FEDEX_INTL_MLSERV";
    ShipmentCarrier["Vamox"] = "VAMOX";
    ShipmentCarrier["AmsGrp"] = "AMS_GRP";
    ShipmentCarrier["DhlJp"] = "DHL_JP";
    ShipmentCarrier["Hrparcel"] = "HRPARCEL";
    ShipmentCarrier["Geswl"] = "GESWL";
    ShipmentCarrier["Bluestar"] = "BLUESTAR";
    ShipmentCarrier["CdekTr"] = "CDEK_TR";
    ShipmentCarrier["Descartes"] = "DESCARTES";
    ShipmentCarrier["DeltecUk"] = "DELTEC_UK";
    ShipmentCarrier["DtdcExpress"] = "DTDC_EXPRESS";
    ShipmentCarrier["Tourline"] = "TOURLINE";
    ShipmentCarrier["BhWorldwide"] = "BH_WORLDWIDE";
    ShipmentCarrier["Ocs"] = "OCS";
    ShipmentCarrier["YingnuoLogistics"] = "YINGNUO_LOGISTICS";
    ShipmentCarrier["Ups"] = "UPS";
    ShipmentCarrier["Toll"] = "TOLL";
    ShipmentCarrier["PrtSeur"] = "PRT_SEUR";
    ShipmentCarrier["DtdcAu"] = "DTDC_AU";
    ShipmentCarrier["ThaDynamicLogistics"] = "THA_DYNAMIC_LOGISTICS";
    ShipmentCarrier["UbiLogistics"] = "UBI_LOGISTICS";
    ShipmentCarrier["FedexCrossborder"] = "FEDEX_CROSSBORDER";
    ShipmentCarrier["A1Post"] = "A1POST";
    ShipmentCarrier["TazmanianFreight"] = "TAZMANIAN_FREIGHT";
    ShipmentCarrier["CjIntMy"] = "CJ_INT_MY";
    ShipmentCarrier["SaiaFreight"] = "SAIA_FREIGHT";
    ShipmentCarrier["SgQxpress"] = "SG_QXPRESS";
    ShipmentCarrier["NhansSolutions"] = "NHANS_SOLUTIONS";
    ShipmentCarrier["DpdFr"] = "DPD_FR";
    ShipmentCarrier["Coordinadora"] = "COORDINADORA";
    ShipmentCarrier["Andreani"] = "ANDREANI";
    ShipmentCarrier["Doora"] = "DOORA";
    ShipmentCarrier["InterparcelNz"] = "INTERPARCEL_NZ";
    ShipmentCarrier["PhlJamexpress"] = "PHL_JAMEXPRESS";
    ShipmentCarrier["BelBelgiumPost"] = "BEL_BELGIUM_POST";
    ShipmentCarrier["UsApc"] = "US_APC";
    ShipmentCarrier["IdnPos"] = "IDN_POS";
    ShipmentCarrier["FrMondial"] = "FR_MONDIAL";
    ShipmentCarrier["DeDhl"] = "DE_DHL";
    ShipmentCarrier["HkRpx"] = "HK_RPX";
    ShipmentCarrier["DhlPieceid"] = "DHL_PIECEID";
    ShipmentCarrier["VnpostEms"] = "VNPOST_EMS";
    ShipmentCarrier["Rrdonnelley"] = "RRDONNELLEY";
    ShipmentCarrier["DpdDe"] = "DPD_DE";
    ShipmentCarrier["DelcartIn"] = "DELCART_IN";
    ShipmentCarrier["Imexglobalsolutions"] = "IMEXGLOBALSOLUTIONS";
    ShipmentCarrier["Acommerce"] = "ACOMMERCE";
    ShipmentCarrier["Eurodis"] = "EURODIS";
    ShipmentCarrier["Canpar"] = "CANPAR";
    ShipmentCarrier["Gls"] = "GLS";
    ShipmentCarrier["IndEcom"] = "IND_ECOM";
    ShipmentCarrier["EspEnvialia"] = "ESP_ENVIALIA";
    ShipmentCarrier["DhlUk"] = "DHL_UK";
    ShipmentCarrier["SmsaExpress"] = "SMSA_EXPRESS";
    ShipmentCarrier["TntFr"] = "TNT_FR";
    ShipmentCarrier["DexI"] = "DEX_I";
    ShipmentCarrier["BudbeeWebhook"] = "BUDBEE_WEBHOOK";
    ShipmentCarrier["CopaCourier"] = "COPA_COURIER";
    ShipmentCarrier["VnmVietnamPost"] = "VNM_VIETNAM_POST";
    ShipmentCarrier["DpdHk"] = "DPD_HK";
    ShipmentCarrier["TollNz"] = "TOLL_NZ";
    ShipmentCarrier["Echo"] = "ECHO";
    ShipmentCarrier["FedexFr"] = "FEDEX_FR";
    ShipmentCarrier["Borderexpress"] = "BORDEREXPRESS";
    ShipmentCarrier["MailplusJpn"] = "MAILPLUS_JPN";
    ShipmentCarrier["TntUkRefr"] = "TNT_UK_REFR";
    ShipmentCarrier["Kec"] = "KEC";
    ShipmentCarrier["DpdRo"] = "DPD_RO";
    ShipmentCarrier["TntJp"] = "TNT_JP";
    ShipmentCarrier["ThCj"] = "TH_CJ";
    ShipmentCarrier["EcCn"] = "EC_CN";
    ShipmentCarrier["FastwayUk"] = "FASTWAY_UK";
    ShipmentCarrier["FastwayUs"] = "FASTWAY_US";
    ShipmentCarrier["GlsDe"] = "GLS_DE";
    ShipmentCarrier["GlsEs"] = "GLS_ES";
    ShipmentCarrier["GlsFr"] = "GLS_FR";
    ShipmentCarrier["MondialBe"] = "MONDIAL_BE";
    ShipmentCarrier["SgtIt"] = "SGT_IT";
    ShipmentCarrier["TntCn"] = "TNT_CN";
    ShipmentCarrier["TntDe"] = "TNT_DE";
    ShipmentCarrier["TntEs"] = "TNT_ES";
    ShipmentCarrier["TntPl"] = "TNT_PL";
    ShipmentCarrier["Parcelforce"] = "PARCELFORCE";
    ShipmentCarrier["SwissPost"] = "SWISS_POST";
    ShipmentCarrier["TollIpec"] = "TOLL_IPEC";
    ShipmentCarrier["Air21"] = "AIR_21";
    ShipmentCarrier["Airspeed"] = "AIRSPEED";
    ShipmentCarrier["Bert"] = "BERT";
    ShipmentCarrier["Bluedart"] = "BLUEDART";
    ShipmentCarrier["Collectplus"] = "COLLECTPLUS";
    ShipmentCarrier["Courierplus"] = "COURIERPLUS";
    ShipmentCarrier["CourierPost"] = "COURIER_POST";
    ShipmentCarrier["DhlGlobalMail"] = "DHL_GLOBAL_MAIL";
    ShipmentCarrier["DpdUk"] = "DPD_UK";
    ShipmentCarrier["DeltecDe"] = "DELTEC_DE";
    ShipmentCarrier["DeutscheDe"] = "DEUTSCHE_DE";
    ShipmentCarrier["Dotzot"] = "DOTZOT";
    ShipmentCarrier["EltaGr"] = "ELTA_GR";
    ShipmentCarrier["EmsCn"] = "EMS_CN";
    ShipmentCarrier["Ecargo"] = "ECARGO";
    ShipmentCarrier["Ensenda"] = "ENSENDA";
    ShipmentCarrier["FercamIt"] = "FERCAM_IT";
    ShipmentCarrier["FastwayZa"] = "FASTWAY_ZA";
    ShipmentCarrier["FastwayAu"] = "FASTWAY_AU";
    ShipmentCarrier["FirstLogisitcs"] = "FIRST_LOGISITCS";
    ShipmentCarrier["Geodis"] = "GEODIS";
    ShipmentCarrier["Globegistics"] = "GLOBEGISTICS";
    ShipmentCarrier["Greyhound"] = "GREYHOUND";
    ShipmentCarrier["JetshipMy"] = "JETSHIP_MY";
    ShipmentCarrier["LionParcel"] = "LION_PARCEL";
    ShipmentCarrier["Aeroflash"] = "AEROFLASH";
    ShipmentCarrier["Ontrac"] = "ONTRAC";
    ShipmentCarrier["Sagawa"] = "SAGAWA";
    ShipmentCarrier["Siodemka"] = "SIODEMKA";
    ShipmentCarrier["Startrack"] = "STARTRACK";
    ShipmentCarrier["TntAu"] = "TNT_AU";
    ShipmentCarrier["TntIt"] = "TNT_IT";
    ShipmentCarrier["Transmission"] = "TRANSMISSION";
    ShipmentCarrier["Yamato"] = "YAMATO";
    ShipmentCarrier["DhlIt"] = "DHL_IT";
    ShipmentCarrier["DhlAt"] = "DHL_AT";
    ShipmentCarrier["LogisticsworldwideKr"] = "LOGISTICSWORLDWIDE_KR";
    ShipmentCarrier["GlsSpain"] = "GLS_SPAIN";
    ShipmentCarrier["AmazonUkApi"] = "AMAZON_UK_API";
    ShipmentCarrier["DpdFrReference"] = "DPD_FR_REFERENCE";
    ShipmentCarrier["DhlparcelUk"] = "DHLPARCEL_UK";
    ShipmentCarrier["Megasave"] = "MEGASAVE";
    ShipmentCarrier["Qualitypost"] = "QUALITYPOST";
    ShipmentCarrier["IdsLogistics"] = "IDS_LOGISTICS";
    ShipmentCarrier["Joyingbox1"] = "JOYINGBOX";
    ShipmentCarrier["PantherOrderNumber"] = "PANTHER_ORDER_NUMBER";
    ShipmentCarrier["WatkinsShepard"] = "WATKINS_SHEPARD";
    ShipmentCarrier["Fasttrack"] = "FASTTRACK";
    ShipmentCarrier["UpExpress"] = "UP_EXPRESS";
    ShipmentCarrier["Elogistica"] = "ELOGISTICA";
    ShipmentCarrier["Ecourier"] = "ECOURIER";
    ShipmentCarrier["CjPhilippines"] = "CJ_PHILIPPINES";
    ShipmentCarrier["Speedex"] = "SPEEDEX";
    ShipmentCarrier["Orangeconnex"] = "ORANGECONNEX";
    ShipmentCarrier["Tecor"] = "TECOR";
    ShipmentCarrier["Saee"] = "SAEE";
    ShipmentCarrier["GlsItalyFtp"] = "GLS_ITALY_FTP";
    ShipmentCarrier["Delivere"] = "DELIVERE";
    ShipmentCarrier["Yycom"] = "YYCOM";
    ShipmentCarrier["AdicionalPt"] = "ADICIONAL_PT";
    ShipmentCarrier["Dksh"] = "DKSH";
    ShipmentCarrier["NipponExpressFtp"] = "NIPPON_EXPRESS_FTP";
    ShipmentCarrier["Gols"] = "GOLS";
    ShipmentCarrier["Fujexp"] = "FUJEXP";
    ShipmentCarrier["Qtrack"] = "QTRACK";
    ShipmentCarrier["OmlogisticsApi"] = "OMLOGISTICS_API";
    ShipmentCarrier["Gdpharm"] = "GDPHARM";
    ShipmentCarrier["MisumiCn"] = "MISUMI_CN";
    ShipmentCarrier["AirCanada"] = "AIR_CANADA";
    ShipmentCarrier["City56Webhook"] = "CITY56_WEBHOOK";
    ShipmentCarrier["SagawaApi"] = "SAGAWA_API";
    ShipmentCarrier["Kedaex"] = "KEDAEX";
    ShipmentCarrier["PgeonApi"] = "PGEON_API";
    ShipmentCarrier["Weworldexpress"] = "WEWORLDEXPRESS";
    ShipmentCarrier["JtLogistics"] = "JT_LOGISTICS";
    ShipmentCarrier["Trusk"] = "TRUSK";
    ShipmentCarrier["Viaxpress"] = "VIAXPRESS";
    ShipmentCarrier["DhlSupplychainId"] = "DHL_SUPPLYCHAIN_ID";
    ShipmentCarrier["ZuelligpharmaSftp"] = "ZUELLIGPHARMA_SFTP";
    ShipmentCarrier["Meest"] = "MEEST";
    ShipmentCarrier["TollPriority"] = "TOLL_PRIORITY";
    ShipmentCarrier["MothershipApi"] = "MOTHERSHIP_API";
    ShipmentCarrier["Capital"] = "CAPITAL";
    ShipmentCarrier["EuropaketApi"] = "EUROPAKET_API";
    ShipmentCarrier["Hfd"] = "HFD";
    ShipmentCarrier["TourlineReference"] = "TOURLINE_REFERENCE";
    ShipmentCarrier["GioEcourier"] = "GIO_ECOURIER";
    ShipmentCarrier["CnLogistics"] = "CN_LOGISTICS";
    ShipmentCarrier["Pandion"] = "PANDION";
    ShipmentCarrier["BpostApi"] = "BPOST_API";
    ShipmentCarrier["Passportshipping"] = "PASSPORTSHIPPING";
    ShipmentCarrier["Pakajo"] = "PAKAJO";
    ShipmentCarrier["Dachser"] = "DACHSER";
    ShipmentCarrier["YusenSftp"] = "YUSEN_SFTP";
    ShipmentCarrier["Shyplite"] = "SHYPLITE";
    ShipmentCarrier["Xyy"] = "XYY";
    ShipmentCarrier["Mwd"] = "MWD";
    ShipmentCarrier["Faxecargo"] = "FAXECARGO";
    ShipmentCarrier["Mazet"] = "MAZET";
    ShipmentCarrier["FirstLogisticsApi"] = "FIRST_LOGISTICS_API";
    ShipmentCarrier["SprintPack"] = "SPRINT_PACK";
    ShipmentCarrier["HermesDeFtp"] = "HERMES_DE_FTP";
    ShipmentCarrier["Concise"] = "CONCISE";
    ShipmentCarrier["KerryExpressTwApi"] = "KERRY_EXPRESS_TW_API";
    ShipmentCarrier["Ewe"] = "EWE";
    ShipmentCarrier["Fastdespatch"] = "FASTDESPATCH";
    ShipmentCarrier["AbcustomSftp"] = "ABCUSTOM_SFTP";
    ShipmentCarrier["Chazki"] = "CHAZKI";
    ShipmentCarrier["Shippie"] = "SHIPPIE";
    ShipmentCarrier["GeodisApi"] = "GEODIS_API";
    ShipmentCarrier["NaqelExpress"] = "NAQEL_EXPRESS";
    ShipmentCarrier["PapaWebhook"] = "PAPA_WEBHOOK";
    ShipmentCarrier["Forwardair"] = "FORWARDAIR";
    ShipmentCarrier["DialogoLogisticaApi"] = "DIALOGO_LOGISTICA_API";
    ShipmentCarrier["LalamoveApi"] = "LALAMOVE_API";
    ShipmentCarrier["Tomydoor"] = "TOMYDOOR";
    ShipmentCarrier["KronosWebhook"] = "KRONOS_WEBHOOK";
    ShipmentCarrier["Jtcargo"] = "JTCARGO";
    ShipmentCarrier["TCat"] = "T_CAT";
    ShipmentCarrier["ConciseWebhook"] = "CONCISE_WEBHOOK";
    ShipmentCarrier["TeleportWebhook"] = "TELEPORT_WEBHOOK";
    ShipmentCarrier["CustomcoApi"] = "CUSTOMCO_API";
    ShipmentCarrier["SpxTh"] = "SPX_TH";
    ShipmentCarrier["BolloreLogistics"] = "BOLLORE_LOGISTICS";
    ShipmentCarrier["ClicklinkSftp"] = "CLICKLINK_SFTP";
    ShipmentCarrier["M3Logistics"] = "M3LOGISTICS";
    ShipmentCarrier["VnpostApi"] = "VNPOST_API";
    ShipmentCarrier["AxlehireFtp"] = "AXLEHIRE_FTP";
    ShipmentCarrier["Shadowfax"] = "SHADOWFAX";
    ShipmentCarrier["MyhermesUkApi"] = "MYHERMES_UK_API";
    ShipmentCarrier["Daiichi"] = "DAIICHI";
    ShipmentCarrier["MensajerosurbanosApi"] = "MENSAJEROSURBANOS_API";
    ShipmentCarrier["Polarspeed"] = "POLARSPEED";
    ShipmentCarrier["IdexpressId"] = "IDEXPRESS_ID";
    ShipmentCarrier["Payo"] = "PAYO";
    ShipmentCarrier["WhistlSftp"] = "WHISTL_SFTP";
    ShipmentCarrier["IntexDe"] = "INTEX_DE";
    ShipmentCarrier["Trans2U"] = "TRANS2U";
    ShipmentCarrier["ProductcaregroupSftp"] = "PRODUCTCAREGROUP_SFTP";
    ShipmentCarrier["Bigsmart"] = "BIGSMART";
    ShipmentCarrier["ExpeditorsApiRef"] = "EXPEDITORS_API_REF";
    ShipmentCarrier["AitworldwideApi"] = "AITWORLDWIDE_API";
    ShipmentCarrier["Worldcourier"] = "WORLDCOURIER";
    ShipmentCarrier["Quiqup"] = "QUIQUP";
    ShipmentCarrier["AgedissSftp"] = "AGEDISS_SFTP";
    ShipmentCarrier["AndreaniApi"] = "ANDREANI_API";
    ShipmentCarrier["Crlexpress"] = "CRLEXPRESS";
    ShipmentCarrier["Smartcat"] = "SMARTCAT";
    ShipmentCarrier["Crossflight"] = "CROSSFLIGHT";
    ShipmentCarrier["Procarrier"] = "PROCARRIER";
    ShipmentCarrier["DhlReferenceApi"] = "DHL_REFERENCE_API";
    ShipmentCarrier["SeinoApi"] = "SEINO_API";
    ShipmentCarrier["Wspexpress"] = "WSPEXPRESS";
    ShipmentCarrier["Kronos"] = "KRONOS";
    ShipmentCarrier["TotalExpressApi"] = "TOTAL_EXPRESS_API";
    ShipmentCarrier["Parcll"] = "PARCLL";
    ShipmentCarrier["Xpedigo"] = "XPEDIGO";
    ShipmentCarrier["StarTrackWebhook"] = "STAR_TRACK_WEBHOOK";
    ShipmentCarrier["Gpost"] = "GPOST";
    ShipmentCarrier["Ucs"] = "UCS";
    ShipmentCarrier["Dmfgroup"] = "DMFGROUP";
    ShipmentCarrier["CoordinadoraApi"] = "COORDINADORA_API";
    ShipmentCarrier["Marken"] = "MARKEN";
    ShipmentCarrier["Ntl"] = "NTL";
    ShipmentCarrier["Redjepakketje"] = "REDJEPAKKETJE";
    ShipmentCarrier["AlliedExpressFtp"] = "ALLIED_EXPRESS_FTP";
    ShipmentCarrier["MondialrelayEs"] = "MONDIALRELAY_ES";
    ShipmentCarrier["NaekoFtp"] = "NAEKO_FTP";
    ShipmentCarrier["Mhi"] = "MHI";
    ShipmentCarrier["Shippify"] = "SHIPPIFY";
    ShipmentCarrier["MalcaAmitApi"] = "MALCA_AMIT_API";
    ShipmentCarrier["JtexpressSgApi"] = "JTEXPRESS_SG_API";
    ShipmentCarrier["DachserWeb"] = "DACHSER_WEB";
    ShipmentCarrier["Flightlg"] = "FLIGHTLG";
    ShipmentCarrier["Cago"] = "CAGO";
    ShipmentCarrier["Com1Express"] = "COM1EXPRESS";
    ShipmentCarrier["TonamiFtp"] = "TONAMI_FTP";
    ShipmentCarrier["Packfleet"] = "PACKFLEET";
    ShipmentCarrier["PurolatorInternational"] = "PUROLATOR_INTERNATIONAL";
    ShipmentCarrier["WineshippingWebhook"] = "WINESHIPPING_WEBHOOK";
    ShipmentCarrier["DhlEsSftp"] = "DHL_ES_SFTP";
    ShipmentCarrier["PchomeApi"] = "PCHOME_API";
    ShipmentCarrier["CeskapostaApi"] = "CESKAPOSTA_API";
    ShipmentCarrier["Gorush"] = "GORUSH";
    ShipmentCarrier["Homerunner"] = "HOMERUNNER";
    ShipmentCarrier["AmazonOrder"] = "AMAZON_ORDER";
    ShipmentCarrier["EfwnowApi"] = "EFWNOW_API";
    ShipmentCarrier["CblLogisticaApi"] = "CBL_LOGISTICA_API";
    ShipmentCarrier["Nimbuspost"] = "NIMBUSPOST";
    ShipmentCarrier["LogwinLogistics"] = "LOGWIN_LOGISTICS";
    ShipmentCarrier["NowlogApi"] = "NOWLOG_API";
    ShipmentCarrier["DpdNl"] = "DPD_NL";
    ShipmentCarrier["Godependable"] = "GODEPENDABLE";
    ShipmentCarrier["Esdex"] = "ESDEX";
    ShipmentCarrier["LogisystemsSftp"] = "LOGISYSTEMS_SFTP";
    ShipmentCarrier["Expeditors"] = "EXPEDITORS";
    ShipmentCarrier["SntglobalApi"] = "SNTGLOBAL_API";
    ShipmentCarrier["Shipx"] = "SHIPX";
    ShipmentCarrier["QintlApi"] = "QINTL_API";
    ShipmentCarrier["Packs"] = "PACKS";
    ShipmentCarrier["PostnlInternational"] = "POSTNL_INTERNATIONAL";
    ShipmentCarrier["AmazonEmailPush"] = "AMAZON_EMAIL_PUSH";
    ShipmentCarrier["DhlApi"] = "DHL_API";
    ShipmentCarrier["Spx"] = "SPX";
    ShipmentCarrier["Axlehire"] = "AXLEHIRE";
    ShipmentCarrier["Icscourier"] = "ICSCOURIER";
    ShipmentCarrier["DialogoLogistica"] = "DIALOGO_LOGISTICA";
    ShipmentCarrier["ShunbangExpress"] = "SHUNBANG_EXPRESS";
    ShipmentCarrier["TcsApi"] = "TCS_API";
    ShipmentCarrier["SfExpressCn"] = "SF_EXPRESS_CN";
    ShipmentCarrier["Packeta"] = "PACKETA";
    ShipmentCarrier["SicTeliway"] = "SIC_TELIWAY";
    ShipmentCarrier["MondialrelayFr"] = "MONDIALRELAY_FR";
    ShipmentCarrier["IntimeFtp"] = "INTIME_FTP";
    ShipmentCarrier["JdExpress"] = "JD_EXPRESS";
    ShipmentCarrier["Fastbox"] = "FASTBOX";
    ShipmentCarrier["Patheon"] = "PATHEON";
    ShipmentCarrier["IndiaPost"] = "INDIA_POST";
    ShipmentCarrier["TipsaRef"] = "TIPSA_REF";
    ShipmentCarrier["Ecofreight"] = "ECOFREIGHT";
    ShipmentCarrier["Vox"] = "VOX";
    ShipmentCarrier["DirectfreightAuRef"] = "DIRECTFREIGHT_AU_REF";
    ShipmentCarrier["BesttransportSftp"] = "BESTTRANSPORT_SFTP";
    ShipmentCarrier["AustraliaPostApi"] = "AUSTRALIA_POST_API";
    ShipmentCarrier["FragilepakSftp"] = "FRAGILEPAK_SFTP";
    ShipmentCarrier["Flipxp"] = "FLIPXP";
    ShipmentCarrier["ValueWebhook"] = "VALUE_WEBHOOK";
    ShipmentCarrier["Daeshin"] = "DAESHIN";
    ShipmentCarrier["Sherpa"] = "SHERPA";
    ShipmentCarrier["MwdApi"] = "MWD_API";
    ShipmentCarrier["Smartkargo"] = "SMARTKARGO";
    ShipmentCarrier["DnjExpress"] = "DNJ_EXPRESS";
    ShipmentCarrier["Gopeople"] = "GOPEOPLE";
    ShipmentCarrier["MysendleApi"] = "MYSENDLE_API";
    ShipmentCarrier["AramexApi"] = "ARAMEX_API";
    ShipmentCarrier["Pidge"] = "PIDGE";
    ShipmentCarrier["Thaiparcels"] = "THAIPARCELS";
    ShipmentCarrier["PantherReferenceApi"] = "PANTHER_REFERENCE_API";
    ShipmentCarrier["Postaplus1"] = "POSTAPLUS";
    ShipmentCarrier["Buffalo"] = "BUFFALO";
    ShipmentCarrier["UEnvios"] = "U_ENVIOS";
    ShipmentCarrier["EliteCo"] = "ELITE_CO";
    ShipmentCarrier["RocheInternalSftp"] = "ROCHE_INTERNAL_SFTP";
    ShipmentCarrier["DbschenkerIceland"] = "DBSCHENKER_ICELAND";
    ShipmentCarrier["TntFrReference"] = "TNT_FR_REFERENCE";
    ShipmentCarrier["Newgisticsapi"] = "NEWGISTICSAPI";
    ShipmentCarrier["Glovo"] = "GLOVO";
    ShipmentCarrier["GwlogisApi"] = "GWLOGIS_API";
    ShipmentCarrier["SpreetailApi"] = "SPREETAIL_API";
    ShipmentCarrier["Moova"] = "MOOVA";
    ShipmentCarrier["Plycongroup"] = "PLYCONGROUP";
    ShipmentCarrier["UspsWebhook"] = "USPS_WEBHOOK";
    ShipmentCarrier["Reimaginedelivery"] = "REIMAGINEDELIVERY";
    ShipmentCarrier["EdfFtp"] = "EDF_FTP";
    ShipmentCarrier["Dao365"] = "DAO365";
    ShipmentCarrier["BiocairFtp"] = "BIOCAIR_FTP";
    ShipmentCarrier["RansaWebhook"] = "RANSA_WEBHOOK";
    ShipmentCarrier["Shipxpres"] = "SHIPXPRES";
    ShipmentCarrier["CourantPlusApi"] = "COURANT_PLUS_API";
    ShipmentCarrier["Shipa"] = "SHIPA";
    ShipmentCarrier["Homelogistics"] = "HOMELOGISTICS";
    ShipmentCarrier["Dx"] = "DX";
    ShipmentCarrier["PosteItalianePaccocelere"] = "POSTE_ITALIANE_PACCOCELERE";
    ShipmentCarrier["TollWebhook"] = "TOLL_WEBHOOK";
    ShipmentCarrier["LctbrApi"] = "LCTBR_API";
    ShipmentCarrier["DxFreight"] = "DX_FREIGHT";
    ShipmentCarrier["DhlSftp"] = "DHL_SFTP";
    ShipmentCarrier["Shiprocket"] = "SHIPROCKET";
    ShipmentCarrier["UberWebhook"] = "UBER_WEBHOOK";
    ShipmentCarrier["Statovernight"] = "STATOVERNIGHT";
    ShipmentCarrier["Burd"] = "BURD";
    ShipmentCarrier["Fastship"] = "FASTSHIP";
    ShipmentCarrier["IbventureWebhook"] = "IBVENTURE_WEBHOOK";
    ShipmentCarrier["GatiKweApi"] = "GATI_KWE_API";
    ShipmentCarrier["CryopdpFtp"] = "CRYOPDP_FTP";
    ShipmentCarrier["Hubbed"] = "HUBBED";
    ShipmentCarrier["TipsaApi"] = "TIPSA_API";
    ShipmentCarrier["Araskargo"] = "ARASKARGO";
    ShipmentCarrier["ThijsNl"] = "THIJS_NL";
    ShipmentCarrier["AtshealthcareReference"] = "ATSHEALTHCARE_REFERENCE";
    ShipmentCarrier["Enum99Minutos"] = "99MINUTOS";
    ShipmentCarrier["HellenicPost"] = "HELLENIC_POST";
    ShipmentCarrier["HsmGlobal"] = "HSM_GLOBAL";
    ShipmentCarrier["Mnx"] = "MNX";
    ShipmentCarrier["Nmtransfer"] = "NMTRANSFER";
    ShipmentCarrier["Logysto"] = "LOGYSTO";
    ShipmentCarrier["IndiaPostInt"] = "INDIA_POST_INT";
    ShipmentCarrier["AmazonFbaSwishipIn"] = "AMAZON_FBA_SWISHIP_IN";
    ShipmentCarrier["SrtTransport"] = "SRT_TRANSPORT";
    ShipmentCarrier["Bomi"] = "BOMI";
    ShipmentCarrier["DeliverrSftp"] = "DELIVERR_SFTP";
    ShipmentCarrier["Hsdexpress"] = "HSDEXPRESS";
    ShipmentCarrier["SimpletireWebhook"] = "SIMPLETIRE_WEBHOOK";
    ShipmentCarrier["HunterExpressSftp"] = "HUNTER_EXPRESS_SFTP";
    ShipmentCarrier["UpsApi"] = "UPS_API";
    ShipmentCarrier["WooyoungLogisticsSftp"] = "WOOYOUNG_LOGISTICS_SFTP";
    ShipmentCarrier["PhseApi"] = "PHSE_API";
    ShipmentCarrier["WishEmailPush"] = "WISH_EMAIL_PUSH";
    ShipmentCarrier["Northline"] = "NORTHLINE";
    ShipmentCarrier["Medafrica"] = "MEDAFRICA";
    ShipmentCarrier["DpdAtSftp"] = "DPD_AT_SFTP";
    ShipmentCarrier["Anteraja"] = "ANTERAJA";
    ShipmentCarrier["DhlGlobalForwardingApi"] = "DHL_GLOBAL_FORWARDING_API";
    ShipmentCarrier["LbcexpressApi"] = "LBCEXPRESS_API";
    ShipmentCarrier["Simsglobal"] = "SIMSGLOBAL";
    ShipmentCarrier["Cdldelivers"] = "CDLDELIVERS";
    ShipmentCarrier["Typ"] = "TYP";
    ShipmentCarrier["TestingCourierWebhook"] = "TESTING_COURIER_WEBHOOK";
    ShipmentCarrier["PandagoApi"] = "PANDAGO_API";
    ShipmentCarrier["RoyalMailFtp"] = "ROYAL_MAIL_FTP";
    ShipmentCarrier["Thunderexpress"] = "THUNDEREXPRESS";
    ShipmentCarrier["SecretlabWebhook"] = "SECRETLAB_WEBHOOK";
    ShipmentCarrier["Setel"] = "SETEL";
    ShipmentCarrier["JdWorldwide"] = "JD_WORLDWIDE";
    ShipmentCarrier["DpdRuApi"] = "DPD_RU_API";
    ShipmentCarrier["ArgentsWebhook"] = "ARGENTS_WEBHOOK";
    ShipmentCarrier["Postone"] = "POSTONE";
    ShipmentCarrier["Tusklogistics"] = "TUSKLOGISTICS";
    ShipmentCarrier["RhenusUkApi"] = "RHENUS_UK_API";
    ShipmentCarrier["TaqbinSgApi"] = "TAQBIN_SG_API";
    ShipmentCarrier["InntralogSftp"] = "INNTRALOG_SFTP";
    ShipmentCarrier["Dayross"] = "DAYROSS";
    ShipmentCarrier["CorreosexpressApi"] = "CORREOSEXPRESS_API";
    ShipmentCarrier["InternationalSeurApi"] = "INTERNATIONAL_SEUR_API";
    ShipmentCarrier["YodelApi"] = "YODEL_API";
    ShipmentCarrier["Heroexpress"] = "HEROEXPRESS";
    ShipmentCarrier["DhlSupplychainIn"] = "DHL_SUPPLYCHAIN_IN";
    ShipmentCarrier["UrgentCargus"] = "URGENT_CARGUS";
    ShipmentCarrier["Frontdoorcorp"] = "FRONTDOORCORP";
    ShipmentCarrier["JtexpressPh"] = "JTEXPRESS_PH";
    ShipmentCarrier["ParcelstarsWebhook"] = "PARCELSTARS_WEBHOOK";
    ShipmentCarrier["DpdSkSftp"] = "DPD_SK_SFTP";
    ShipmentCarrier["Movianto"] = "MOVIANTO";
    ShipmentCarrier["OzepartsShipping"] = "OZEPARTS_SHIPPING";
    ShipmentCarrier["Kargomkolay"] = "KARGOMKOLAY";
    ShipmentCarrier["Trunkrs"] = "TRUNKRS";
    ShipmentCarrier["OmnirpsWebhook"] = "OMNIRPS_WEBHOOK";
    ShipmentCarrier["Chilexpress"] = "CHILEXPRESS";
    ShipmentCarrier["TestingCourier"] = "TESTING_COURIER";
    ShipmentCarrier["JneApi"] = "JNE_API";
    ShipmentCarrier["BjshomedeliveryFtp"] = "BJSHOMEDELIVERY_FTP";
    ShipmentCarrier["DexpressWebhook"] = "DEXPRESS_WEBHOOK";
    ShipmentCarrier["UspsApi"] = "USPS_API";
    ShipmentCarrier["Transvirtual"] = "TRANSVIRTUAL";
    ShipmentCarrier["SolisticaApi"] = "SOLISTICA_API";
    ShipmentCarrier["ChienventureWebhook"] = "CHIENVENTURE_WEBHOOK";
    ShipmentCarrier["DpdUkSftp"] = "DPD_UK_SFTP";
    ShipmentCarrier["InpostUk"] = "INPOST_UK";
    ShipmentCarrier["Javit"] = "JAVIT";
    ShipmentCarrier["ZtoDomestic"] = "ZTO_DOMESTIC";
    ShipmentCarrier["DhlGtApi"] = "DHL_GT_API";
    ShipmentCarrier["CevaTracking"] = "CEVA_TRACKING";
    ShipmentCarrier["KomonExpress"] = "KOMON_EXPRESS";
    ShipmentCarrier["EastwestcourierFtp"] = "EASTWESTCOURIER_FTP";
    ShipmentCarrier["Danniao"] = "DANNIAO";
    ShipmentCarrier["Spectran"] = "SPECTRAN";
    ShipmentCarrier["DeliverIt"] = "DELIVER_IT";
    ShipmentCarrier["Relaiscolis"] = "RELAISCOLIS";
    ShipmentCarrier["GlsSpainApi"] = "GLS_SPAIN_API";
    ShipmentCarrier["Postplus"] = "POSTPLUS";
    ShipmentCarrier["Airterra"] = "AIRTERRA";
    ShipmentCarrier["GioEcourierApi"] = "GIO_ECOURIER_API";
    ShipmentCarrier["DpdChSftp"] = "DPD_CH_SFTP";
    ShipmentCarrier["FedexApi"] = "FEDEX_API";
    ShipmentCarrier["Intersmarttrans"] = "INTERSMARTTRANS";
    ShipmentCarrier["HermesUkSftp"] = "HERMES_UK_SFTP";
    ShipmentCarrier["ExelotFtp"] = "EXELOT_FTP";
    ShipmentCarrier["DhlPaApi"] = "DHL_PA_API";
    ShipmentCarrier["VirtransportSftp"] = "VIRTRANSPORT_SFTP";
    ShipmentCarrier["Worldnet"] = "WORLDNET";
    ShipmentCarrier["InstaboxWebhook"] = "INSTABOX_WEBHOOK";
    ShipmentCarrier["Kng"] = "KNG";
    ShipmentCarrier["FlashexpressWebhook"] = "FLASHEXPRESS_WEBHOOK";
    ShipmentCarrier["MagyarPostaApi"] = "MAGYAR_POSTA_API";
    ShipmentCarrier["WeshipApi"] = "WESHIP_API";
    ShipmentCarrier["OhiWebhook"] = "OHI_WEBHOOK";
    ShipmentCarrier["Mudita"] = "MUDITA";
    ShipmentCarrier["BluedartApi"] = "BLUEDART_API";
    ShipmentCarrier["TCatApi"] = "T_CAT_API";
    ShipmentCarrier["Ads"] = "ADS";
    ShipmentCarrier["HermesIt"] = "HERMES_IT";
    ShipmentCarrier["FitzmarkApi"] = "FITZMARK_API";
    ShipmentCarrier["PostiApi"] = "POSTI_API";
    ShipmentCarrier["SmsaExpressWebhook"] = "SMSA_EXPRESS_WEBHOOK";
    ShipmentCarrier["TamergroupWebhook"] = "TAMERGROUP_WEBHOOK";
    ShipmentCarrier["Livrapide"] = "LIVRAPIDE";
    ShipmentCarrier["NipponExpress"] = "NIPPON_EXPRESS";
    ShipmentCarrier["Bettertrucks"] = "BETTERTRUCKS";
    ShipmentCarrier["Fan"] = "FAN";
    ShipmentCarrier["PbUspsflatsFtp"] = "PB_USPSFLATS_FTP";
    ShipmentCarrier["Parcelright"] = "PARCELRIGHT";
    ShipmentCarrier["Ithinklogistics"] = "ITHINKLOGISTICS";
    ShipmentCarrier["KerryExpressThWebhook"] = "KERRY_EXPRESS_TH_WEBHOOK";
    ShipmentCarrier["Ecoutier"] = "ECOUTIER";
    ShipmentCarrier["Showl"] = "SHOWL";
    ShipmentCarrier["BrtItApi"] = "BRT_IT_API";
    ShipmentCarrier["RixonhkApi"] = "RIXONHK_API";
    ShipmentCarrier["DbschenkerApi"] = "DBSCHENKER_API";
    ShipmentCarrier["Ilyanglogis"] = "ILYANGLOGIS";
    ShipmentCarrier["MailBoxEtc"] = "MAIL_BOX_ETC";
    ShipmentCarrier["Weship"] = "WESHIP";
    ShipmentCarrier["DhlGlobalMailApi"] = "DHL_GLOBAL_MAIL_API";
    ShipmentCarrier["Activos24Api"] = "ACTIVOS24_API";
    ShipmentCarrier["Atshealthcare"] = "ATSHEALTHCARE";
    ShipmentCarrier["Luwjistik"] = "LUWJISTIK";
    ShipmentCarrier["GwWorld"] = "GW_WORLD";
    ShipmentCarrier["FairsendenApi"] = "FAIRSENDEN_API";
    ShipmentCarrier["ServipWebhook"] = "SERVIP_WEBHOOK";
    ShipmentCarrier["Swiship"] = "SWISHIP";
    ShipmentCarrier["Tanet"] = "TANET";
    ShipmentCarrier["HotsinCargo"] = "HOTSIN_CARGO";
    ShipmentCarrier["Direx"] = "DIREX";
    ShipmentCarrier["Huantong"] = "HUANTONG";
    ShipmentCarrier["ImileApi"] = "IMILE_API";
    ShipmentCarrier["Auexpress"] = "AUEXPRESS";
    ShipmentCarrier["Nytlogistics"] = "NYTLOGISTICS";
    ShipmentCarrier["DsvReference"] = "DSV_REFERENCE";
    ShipmentCarrier["NovofarmaWebhook"] = "NOVOFARMA_WEBHOOK";
    ShipmentCarrier["AitworldwideSftp"] = "AITWORLDWIDE_SFTP";
    ShipmentCarrier["Shopolive"] = "SHOPOLIVE";
    ShipmentCarrier["FnfZa"] = "FNF_ZA";
    ShipmentCarrier["DhlEcommerceGc"] = "DHL_ECOMMERCE_GC";
    ShipmentCarrier["Fetchr"] = "FETCHR";
    ShipmentCarrier["StarlinksApi"] = "STARLINKS_API";
    ShipmentCarrier["Yyexpress"] = "YYEXPRESS";
    ShipmentCarrier["Servientrega"] = "SERVIENTREGA";
    ShipmentCarrier["Hanjin"] = "HANJIN";
    ShipmentCarrier["SpanishSeurFtp"] = "SPANISH_SEUR_FTP";
    ShipmentCarrier["DxB2BConnum"] = "DX_B2B_CONNUM";
    ShipmentCarrier["HelthjemApi"] = "HELTHJEM_API";
    ShipmentCarrier["Inexpost"] = "INEXPOST";
    ShipmentCarrier["A2BBa"] = "A2B_BA";
    ShipmentCarrier["RhenusGroup"] = "RHENUS_GROUP";
    ShipmentCarrier["SberlogisticsRu"] = "SBERLOGISTICS_RU";
    ShipmentCarrier["MalcaAmit"] = "MALCA_AMIT";
    ShipmentCarrier["Ppl"] = "PPL";
    ShipmentCarrier["OsmWorldwideSftp"] = "OSM_WORLDWIDE_SFTP";
    ShipmentCarrier["Acilogistix"] = "ACILOGISTIX";
    ShipmentCarrier["Optimacourier"] = "OPTIMACOURIER";
    ShipmentCarrier["NovaPoshtaApi"] = "NOVA_POSHTA_API";
    ShipmentCarrier["Loggi"] = "LOGGI";
    ShipmentCarrier["Yifan"] = "YIFAN";
    ShipmentCarrier["Mydynalogic"] = "MYDYNALOGIC";
    ShipmentCarrier["Morninglobal"] = "MORNINGLOBAL";
    ShipmentCarrier["ConciseApi"] = "CONCISE_API";
    ShipmentCarrier["Fxtran"] = "FXTRAN";
    ShipmentCarrier["DeliveryourparcelZa"] = "DELIVERYOURPARCEL_ZA";
    ShipmentCarrier["Uparcel"] = "UPARCEL";
    ShipmentCarrier["MobiBr"] = "MOBI_BR";
    ShipmentCarrier["LoginextWebhook"] = "LOGINEXT_WEBHOOK";
    ShipmentCarrier["Ems"] = "EMS";
    ShipmentCarrier["Speedy"] = "SPEEDY";
    ShipmentCarrier["ZoomRed"] = "ZOOM_RED";
    ShipmentCarrier["Navlungo"] = "NAVLUNGO";
    ShipmentCarrier["Castleparcels"] = "CASTLEPARCELS";
    ShipmentCarrier["Weee"] = "WEEE";
    ShipmentCarrier["Packaly"] = "PACKALY";
    ShipmentCarrier["Yunhuipost"] = "YUNHUIPOST";
    ShipmentCarrier["Youparcel"] = "YOUPARCEL";
    ShipmentCarrier["Leman"] = "LEMAN";
    ShipmentCarrier["Moovin"] = "MOOVIN";
    ShipmentCarrier["UrbIt"] = "URB_IT";
    ShipmentCarrier["Multientregapanama"] = "MULTIENTREGAPANAMA";
    ShipmentCarrier["Jusdasr"] = "JUSDASR";
    ShipmentCarrier["Discountpost"] = "DISCOUNTPOST";
    ShipmentCarrier["RhenusUk"] = "RHENUS_UK";
    ShipmentCarrier["SwishipJp"] = "SWISHIP_JP";
    ShipmentCarrier["GlsUs"] = "GLS_US";
    ShipmentCarrier["Smtl"] = "SMTL";
    ShipmentCarrier["Emega"] = "EMEGA";
    ShipmentCarrier["ExpressoneSv"] = "EXPRESSONE_SV";
    ShipmentCarrier["Hepsijet"] = "HEPSIJET";
    ShipmentCarrier["Welivery"] = "WELIVERY";
    ShipmentCarrier["Bringer"] = "BRINGER";
    ShipmentCarrier["Easyroutes"] = "EASYROUTES";
    ShipmentCarrier["Mrw"] = "MRW";
    ShipmentCarrier["Rpm"] = "RPM";
    ShipmentCarrier["DpdPrt"] = "DPD_PRT";
    ShipmentCarrier["GlsRomania"] = "GLS_ROMANIA";
    ShipmentCarrier["Lmparcel"] = "LMPARCEL";
    ShipmentCarrier["Gtagsm"] = "GTAGSM";
    ShipmentCarrier["Domino"] = "DOMINO";
    ShipmentCarrier["Eshipper"] = "ESHIPPER";
    ShipmentCarrier["Transpak"] = "TRANSPAK";
    ShipmentCarrier["Xindus"] = "XINDUS";
    ShipmentCarrier["Aoyue"] = "AOYUE";
    ShipmentCarrier["Easyparcel"] = "EASYPARCEL";
    ShipmentCarrier["Expressone"] = "EXPRESSONE";
    ShipmentCarrier["SendeoKargo"] = "SENDEO_KARGO";
    ShipmentCarrier["Speedaf"] = "SPEEDAF";
    ShipmentCarrier["Etower"] = "ETOWER";
    ShipmentCarrier["Gcx"] = "GCX";
    ShipmentCarrier["NinjavanVn"] = "NINJAVAN_VN";
    ShipmentCarrier["Allegro"] = "ALLEGRO";
    ShipmentCarrier["Jumppoint"] = "JUMPPOINT";
    ShipmentCarrier["ShipglobalUs"] = "SHIPGLOBAL_US";
    ShipmentCarrier["Kinisi"] = "KINISI";
    ShipmentCarrier["Oakh"] = "OAKH";
    ShipmentCarrier["Awest"] = "AWEST";
    ShipmentCarrier["Barsan"] = "BARSAN";
    ShipmentCarrier["Energologistic"] = "ENERGOLOGISTIC";
    ShipmentCarrier["Madrooex"] = "MADROOEX";
    ShipmentCarrier["Gobolt"] = "GOBOLT";
    ShipmentCarrier["SwissUniversalExpress"] = "SWISS_UNIVERSAL_EXPRESS";
    ShipmentCarrier["Iordirect"] = "IORDIRECT";
    ShipmentCarrier["Xmszm"] = "XMSZM";
    ShipmentCarrier["GlsHun"] = "GLS_HUN";
    ShipmentCarrier["Sendy"] = "SENDY";
    ShipmentCarrier["Braunsexpress"] = "BRAUNSEXPRESS";
    ShipmentCarrier["Grandslamexpress"] = "GRANDSLAMEXPRESS";
    ShipmentCarrier["Xgs"] = "XGS";
    ShipmentCarrier["Otschile"] = "OTSCHILE";
    ShipmentCarrier["PackUp"] = "PACK_UP";
    ShipmentCarrier["Parcelstars"] = "PARCELSTARS";
    ShipmentCarrier["Teamexpressllc"] = "TEAMEXPRESSLLC";
    ShipmentCarrier["Asyadexpress"] = "ASYADEXPRESS";
    ShipmentCarrier["Tdn"] = "TDN";
    ShipmentCarrier["Earlybird"] = "EARLYBIRD";
    ShipmentCarrier["Cacesa"] = "CACESA";
    ShipmentCarrier["Parceljet"] = "PARCELJET";
    ShipmentCarrier["MngKargo"] = "MNG_KARGO";
    ShipmentCarrier["Superpackline"] = "SUPERPACKLINE";
    ShipmentCarrier["Speedx"] = "SPEEDX";
    ShipmentCarrier["Vesyl"] = "VESYL";
    ShipmentCarrier["Skyking"] = "SKYKING";
    ShipmentCarrier["Dirmensajeria"] = "DIRMENSAJERIA";
    ShipmentCarrier["Netlogixgroup"] = "NETLOGIXGROUP";
    ShipmentCarrier["Zyou"] = "ZYOU";
    ShipmentCarrier["Jawar"] = "JAWAR";
    ShipmentCarrier["Agsystems"] = "AGSYSTEMS";
    ShipmentCarrier["Gps"] = "GPS";
    ShipmentCarrier["PttKargo"] = "PTT_KARGO";
    ShipmentCarrier["Maergo"] = "MAERGO";
    ShipmentCarrier["Arihantcourier"] = "ARIHANTCOURIER";
    ShipmentCarrier["Vtfe"] = "VTFE";
    ShipmentCarrier["Yunant"] = "YUNANT";
    ShipmentCarrier["Urbify"] = "URBIFY";
    ShipmentCarrier["PackMan"] = "PACK_MAN";
    ShipmentCarrier["Liefergrun"] = "LIEFERGRUN";
    ShipmentCarrier["Obibox"] = "OBIBOX";
    ShipmentCarrier["Paikeda"] = "PAIKEDA";
    ShipmentCarrier["Scotty"] = "SCOTTY";
    ShipmentCarrier["IntelcomCa"] = "INTELCOM_CA";
    ShipmentCarrier["Swe"] = "SWE";
    ShipmentCarrier["Asendia"] = "ASENDIA";
    ShipmentCarrier["DpdAt"] = "DPD_AT";
    ShipmentCarrier["Relay"] = "RELAY";
    ShipmentCarrier["Ata"] = "ATA";
    ShipmentCarrier["SkyexpressInternational"] = "SKYEXPRESS_INTERNATIONAL";
    ShipmentCarrier["SuratKargo"] = "SURAT_KARGO";
    ShipmentCarrier["Sglink"] = "SGLINK";
    ShipmentCarrier["Fleetopticsinc"] = "FLEETOPTICSINC";
    ShipmentCarrier["Shopline"] = "SHOPLINE";
    ShipmentCarrier["Piggyship"] = "PIGGYSHIP";
    ShipmentCarrier["Logoix"] = "LOGOIX";
    ShipmentCarrier["KolayGelsin"] = "KOLAY_GELSIN";
    ShipmentCarrier["AssociatedCouriers"] = "ASSOCIATED_COURIERS";
    ShipmentCarrier["UpsChecker"] = "UPS_CHECKER";
    ShipmentCarrier["Wineshipping"] = "WINESHIPPING";
    ShipmentCarrier["Spedisci"] = "SPEDISCI";
})(ShipmentCarrier = exports.ShipmentCarrier || (exports.ShipmentCarrier = {}));
/**
 * Schema for ShipmentCarrier
 */
exports.shipmentCarrierSchema = (0, schema_1.stringEnum)(ShipmentCarrier, true);
