// @paypal/paypal-server-sdk/dist/cjs/models/shippingDetails.js
"use strict";
/**
 * PayPal Server SDKLib
 *
 * This file was automatically generated by APIMATIC v3.0 ( https://www.apimatic.io ).
 */
exports.__esModule = true;
exports.shippingDetailsSchema = void 0;
var schema_1 = require("../schema");
var address_1 = require("./address");
var fulfillmentType_1 = require("./fulfillmentType");
var phoneNumberWithCountryCode_1 = require("./phoneNumberWithCountryCode");
var shippingName_1 = require("./shippingName");
var shippingOption_1 = require("./shippingOption");
exports.shippingDetailsSchema = (0, schema_1.object)({
    name: ['name', (0, schema_1.optional)((0, schema_1.lazy)(function () { return shippingName_1.shippingNameSchema; }))],
    emailAddress: ['email_address', (0, schema_1.optional)((0, schema_1.string)())],
    phoneNumber: [
        'phone_number',
        (0, schema_1.optional)((0, schema_1.lazy)(function () { return phoneNumberWithCountryCode_1.phoneNumberWithCountryCodeSchema; })),
    ],
    type: ['type', (0, schema_1.optional)(fulfillmentType_1.fulfillmentTypeSchema)],
    options: ['options', (0, schema_1.optional)((0, schema_1.array)((0, schema_1.lazy)(function () { return shippingOption_1.shippingOptionSchema; })))],
    address: ['address', (0, schema_1.optional)((0, schema_1.lazy)(function () { return address_1.addressSchema; }))]
});
