// @paypal/paypal-server-sdk/dist/cjs/models/usagePattern.js
"use strict";
/**
 * PayPal Server SDKLib
 *
 * This file was automatically generated by APIMATIC v3.0 ( https://www.apimatic.io ).
 */
exports.__esModule = true;
exports.usagePatternSchema = exports.UsagePattern = void 0;
var schema_1 = require("../schema");
/**
 * Enum for UsagePattern
 */
var UsagePattern;
(function (UsagePattern) {
    UsagePattern["Immediate"] = "IMMEDIATE";
    UsagePattern["Deferred"] = "DEFERRED";
    UsagePattern["RecurringPrepaid"] = "RECURRING_PREPAID";
    UsagePattern["RecurringPostpaid"] = "RECURRING_POSTPAID";
    UsagePattern["ThresholdPrepaid"] = "THRESHOLD_PREPAID";
    UsagePattern["ThresholdPostpaid"] = "THRESHOLD_POSTPAID";
    UsagePattern["SubscriptionPrepaid"] = "SUBSCRIPTION_PREPAID";
    UsagePattern["SubscriptionPostpaid"] = "SUBSCRIPTION_POSTPAID";
    UsagePattern["UnscheduledPrepaid"] = "UNSCHEDULED_PREPAID";
    UsagePattern["UnscheduledPostpaid"] = "UNSCHEDULED_POSTPAID";
    UsagePattern["InstallmentPrepaid"] = "INSTALLMENT_PREPAID";
    UsagePattern["InstallmentPostpaid"] = "INSTALLMENT_POSTPAID";
})(UsagePattern = exports.UsagePattern || (exports.UsagePattern = {}));
/**
 * Schema for UsagePattern
 */
exports.usagePatternSchema = (0, schema_1.stringEnum)(UsagePattern, true);
