// @paypal/paypal-server-sdk/dist/cjs/models/vaultExperienceContext.js
"use strict";
/**
 * PayPal Server SDKLib
 *
 * This file was automatically generated by APIMATIC v3.0 ( https://www.apimatic.io ).
 */
exports.__esModule = true;
exports.vaultExperienceContextSchema = void 0;
var schema_1 = require("../schema");
var orderApplicationContextShippingPreference_1 = require("./orderApplicationContextShippingPreference");
var vaultInstructionAction_1 = require("./vaultInstructionAction");
exports.vaultExperienceContextSchema = (0, schema_1.object)({
    brandName: ['brand_name', (0, schema_1.optional)((0, schema_1.string)())],
    locale: ['locale', (0, schema_1.optional)((0, schema_1.string)())],
    returnUrl: ['return_url', (0, schema_1.optional)((0, schema_1.string)())],
    cancelUrl: ['cancel_url', (0, schema_1.optional)((0, schema_1.string)())],
    shippingPreference: [
        'shipping_preference',
        (0, schema_1.optional)(orderApplicationContextShippingPreference_1.orderApplicationContextShippingPreferenceSchema),
    ],
    vaultInstruction: [
        'vault_instruction',
        (0, schema_1.optional)(vaultInstructionAction_1.vaultInstructionActionSchema),
    ]
});
