// @paypal/paypal-server-sdk/dist/cjs/models/vaultPaypalWalletRequest.js
"use strict";
/**
 * PayPal Server SDKLib
 *
 * This file was automatically generated by APIMATIC v3.0 ( https://www.apimatic.io ).
 */
exports.__esModule = true;
exports.vaultPaypalWalletRequestSchema = void 0;
var schema_1 = require("../schema");
var paypalPaymentTokenCustomerType_1 = require("./paypalPaymentTokenCustomerType");
var paypalPaymentTokenUsageType_1 = require("./paypalPaymentTokenUsageType");
var plan_1 = require("./plan");
var usagePattern_1 = require("./usagePattern");
var vaultedDigitalWalletShippingDetails_1 = require("./vaultedDigitalWalletShippingDetails");
var vaultExperienceContext_1 = require("./vaultExperienceContext");
exports.vaultPaypalWalletRequestSchema = (0, schema_1.object)({
    description: ['description', (0, schema_1.optional)((0, schema_1.string)())],
    usagePattern: ['usage_pattern', (0, schema_1.optional)(usagePattern_1.usagePatternSchema)],
    shipping: [
        'shipping',
        (0, schema_1.optional)((0, schema_1.lazy)(function () { return vaultedDigitalWalletShippingDetails_1.vaultedDigitalWalletShippingDetailsSchema; })),
    ],
    permitMultiplePaymentTokens: [
        'permit_multiple_payment_tokens',
        (0, schema_1.optional)((0, schema_1.boolean)()),
    ],
    usageType: ['usage_type', (0, schema_1.optional)(paypalPaymentTokenUsageType_1.paypalPaymentTokenUsageTypeSchema)],
    customerType: [
        'customer_type',
        (0, schema_1.optional)(paypalPaymentTokenCustomerType_1.paypalPaymentTokenCustomerTypeSchema),
    ],
    billingPlan: ['billing_plan', (0, schema_1.optional)((0, schema_1.lazy)(function () { return plan_1.planSchema; }))],
    experienceContext: [
        'experience_context',
        (0, schema_1.optional)((0, schema_1.lazy)(function () { return vaultExperienceContext_1.vaultExperienceContextSchema; })),
    ]
});
