// @paypal/paypal-server-sdk/dist/cjs/models/venmoWalletResponse.js
"use strict";
/**
 * PayPal Server SDKLib
 *
 * This file was automatically generated by APIMATIC v3.0 ( https://www.apimatic.io ).
 */
exports.__esModule = true;
exports.venmoWalletResponseSchema = void 0;
var schema_1 = require("../schema");
var address_1 = require("./address");
var name_1 = require("./name");
var phoneNumber_1 = require("./phoneNumber");
var venmoWalletAttributesResponse_1 = require("./venmoWalletAttributesResponse");
exports.venmoWalletResponseSchema = (0, schema_1.object)({
    emailAddress: ['email_address', (0, schema_1.optional)((0, schema_1.string)())],
    accountId: ['account_id', (0, schema_1.optional)((0, schema_1.string)())],
    userName: ['user_name', (0, schema_1.optional)((0, schema_1.string)())],
    name: ['name', (0, schema_1.optional)((0, schema_1.lazy)(function () { return name_1.nameSchema; }))],
    phoneNumber: ['phone_number', (0, schema_1.optional)((0, schema_1.lazy)(function () { return phoneNumber_1.phoneNumberSchema; }))],
    address: ['address', (0, schema_1.optional)((0, schema_1.lazy)(function () { return address_1.addressSchema; }))],
    attributes: [
        'attributes',
        (0, schema_1.optional)((0, schema_1.lazy)(function () { return venmoWalletAttributesResponse_1.venmoWalletAttributesResponseSchema; })),
    ]
});
